---
title: "16S"
author: "Nicholas Baetge"
date: "8/13/2020"
output: github_document
---

# Intro

Here, the NAAMES cast 16S sequences processed by Luis Bolanos in the Giovannoni group are explored

```{r load packages, message=FALSE, warning=FALSE}
library(tidyverse) 
library(rmarkdown)
library(patchwork)
library(lubridate)
library(phyloseq)
library(RColorBrewer)
library(ggpubr)
library(viridis)
```


```{r aesthetics}
custom.colors <- c("AT39" = "#377EB8", "AT34" = "#4DAF4A", "AT38" = "#E41A1C", "AT32" = "#FF7F00", "NAAMES 4" = "#377EB8", "NAAMES 2" = "#4DAF4A", "NAAMES 3" = "#E41A1C", "NAAMES 1" = "#FF7F00", "Temperate" = "#A6CEE3", "Subpolar" = "#377EB8", "Subtropical" = "#FB9A99", "GS/Sargasso" = "#E41A1C", "Early Spring" = "#377EB8", "Late Spring" = "#4DAF4A","Early Autumn" = "#E41A1C", "Late Autumn" = "#FF7F00",  "0-75 m" = "#A50026", "75-200 m" = "#313695", "Euphotic" = "#A50026", "Upper Mesopelagic" = "#313695", "Mesopelagic" = "#d16cfa")

levels = c("GS/Sargasso", "Subtropical", "Temperate", "Subpolar",  "AT39-6", "AT34", "AT38", "AT32","South", "North", "Early Spring", "Late Spring","Early Autumn",  "Late Autumn", "5-75 m", "100-200 m", "300 m", "Euphotic", "Upper Mesopelagic")

#odv.colors <- c("#feb483", "#d31f2a", "#ffc000", "#27ab19", "#0db5e6", "#7139fe", "#d16cfa")
odv.colors <- c( "#d31f2a", "#ffc000", "#27ab19", "#0db5e6", "#7139fe", "#d16cfa")

matlab.colors <- c("#A50026", "#D73027", "#F46D43", "#FDAE61", "#FEE090", "#FFFFBF", "#E0F3F8", "#ABD9E9", "#74ADD1", "#4575B4", "#313695")
matlab.colors2 <- c("#A50026", "#D73027", "#F46D43", "#FDAE61",  "#ABD9E9", "#74ADD1", "#4575B4", "#313695")
```

# Import Data 

```{r load data, message=FALSE, warning=FALSE}

count.tab <- read.table("~/GITHUB/naames_multiday/Input/16s/HetV1OTU.txt", header = T, row.names = 1, check.names = F) 

tax.tab <- as.matrix(read.table("~/GITHUB/naames_multiday/Input/16s/HetV1TUtax.txt", header = T, row.names = 1, check.names = F, na.strings = "", sep = "\t"))


ctd <-  readRDS("~/GITHUB/naames_multiday/Input/ctd_data.rds") %>%
  select(Cruise, Station, CampCN,  z, deriv_o2_umol_l, aou_c, fl_mg_m3, ave_temp_c, ave_sal_psu, ave_dens_kg_m3, beamT_perc ) %>%
  mutate(Cruise = ifelse(Cruise == "AT39", "AT39-6", Cruise)) %>%
  rename(o2 = deriv_o2_umol_l,
         aou_c = aou_c,
         fl = fl_mg_m3,
         temp = ave_temp_c,
         sal = ave_sal_psu,
         dens = ave_dens_kg_m3,
         beamt = beamT_perc) %>% 
  mutate_at(vars(Station), as.character) %>% 
  mutate(Station = ifelse(Cruise == "AT38" & Station == 0, "1A", Station)) 
  


npp <- read_rds("~/GITHUB/naames_multiday/Input/npp_data.rds")

sample.tab <- read_rds("~/GITHUB/naames_multiday/Input/master/processed_bf.8.2020.rds") %>% 
  select(Cruise:CampCN, Target_Z, DNA_ID) %>% 
  drop_na(DNA_ID) %>% 
  rename(z = Target_Z) %>% 
  left_join(., read_rds("~/GITHUB/naames_multiday/Output/processed_data.rds") %>%
              select(Cruise, Station, Date,  CampCN, mld, ez, z, ba, doc, tdaa, n, phyc, bcd ) %>% 
              distinct() %>% 
              mutate_at(vars(Station), as.character) %>% 
              mutate(Station = ifelse(Cruise == "AT38" & Station == 0, "1A", Station)) %>% 
              mutate_at(vars(phyc:bcd), function(x)(x/10^3))) %>% 
  mutate(z_interv = ifelse(z <= 75, "Euphotic", NA),
         z_interv = ifelse(z >= 100 & z <= 200 , "Upper Mesopelagic", z_interv),
         z_interv = ifelse(z == 300, "300 m", z_interv),
    
         ) %>% 
  select(Cruise:z,  z_interv, everything()) %>% 
  mutate(Cruise2 = ifelse(Cruise == "AT32", "NAAMES 1", "NAAMES 2"),
         Cruise2 = ifelse(Cruise == "AT38", "NAAMES 3", Cruise2),
         Cruise2 = ifelse(Cruise == "AT39", "NAAMES 4", Cruise2)) %>% 
  select(Cruise, Cruise2, everything()) %>% 
  left_join(., ctd) %>% 
  left_join(., npp %>% mutate_at(vars(Station), as.character)) %>% 
  rename(latitude = Latitude) %>% 
  mutate(eddy = ifelse(Cruise == "AT34" & Station == 4, "Core", NA),
         eddy = ifelse(Date == "2016-05-27", "Periphery", eddy)) %>% 
  group_by(Cruise, Station) %>% 
  mutate(time = ymd_hms(datetime),
         interv = interval(first(time), time),
         dur = as.duration(interv),
         days = round(as.numeric(dur, "days"))) %>% 
  ungroup() %>% 
  column_to_rownames(var = "DNA_ID") %>% 
  select(Cruise:CampCN, eddy,  days, everything()) %>% 
  select(-c(time:dur)) 

gene_copies <- readxl::read_xlsx("~/GITHUB/naames_multiday/Input/16s/gene_copies.xlsx") %>% 
  mutate(copy_num = ave_order, 
         copy_num = ifelse(is.na(copy_num), ave_class, copy_num),
         copy_num = ifelse(is.na(copy_num), ave_phylum, copy_num),
         copy_num = ifelse(is.na(copy_num), ave_kingdom, copy_num))

```

# Phyloseq Object

We need to create a phyloseq object that merges all three datasets. Sometimes this doesn't work beacuse of the format of the data files. Make sure all the sample names between the sampleinfo.txt and seqtab-nochimtaxa.txt are the same

```{r}
OTU = otu_table(count.tab, taxa_are_rows = TRUE) 
TAX = tax_table(tax.tab)
SAM = sample_data(sample.tab)
ps = phyloseq(OTU,TAX,SAM) 

sample_data(ps)$z_interv <- factor(sample_data(ps)$z_interv, levels = levels)
sample_data(ps)$Season <- factor(sample_data(ps)$Season, levels = levels)

```

# Filter sequences

We will filter out  chloroplasts and mitochondria, because we only intended to amplify bacterial sequences. It's good to check you don’t have anything lurking in the taxonomy table. 

```{r}
sub_ps <- ps %>%
  subset_samples(z %in% c(5, 25, 50, 75, 100, 150, 200)) %>%
  subset_taxa(
    Family  != "mitochondria" &
    Order   != "Chloroplast")
```

# Sample Summary

As a first analysis, we will look at the distribution of read counts from our samples

```{r echo = FALSE, warning = FALSE, message = FALSE, fig.height = 6, fig.width = 12, fig.align = "center"}
# Make a data frame with a column for the read counts of each sample
sample_sum_df <- data.frame(sum = sample_sums(sub_ps))

# Histogram of sample read counts
ggplot(sample_sum_df, aes(x = sum)) + 
  geom_histogram(color = "black", fill = "#377EB8", binwidth = 2500) +
  ggtitle("Distribution of sample sequencing depth") + 
  xlab("Read counts") +
  theme(axis.title.y = element_blank()) +
  theme_classic2(base_size = 16)
```

```{r}
# mean, max and min of sample read counts
smin <- min(sample_sums(sub_ps)) 
smean <- mean(sample_sums(sub_ps)) 
smax <- max(sample_sums(sub_ps)) 
```

# Beta Diversity

Beta diversity involves calculating metrics such as distances or dissimilarities based on pairwise comparisons of samples – they don’t exist for a single sample, but rather only as metrics that relate samples to each other. i.e. beta diversity =  patterns in community structure between samples

Since differences in sampling depths between samples can influence distance/dissimilarity metrics, we first need to somehow normalize the read depth across our samples. 

## Subsample

We will rarefy (random subsample with replacement) the min read depth of the samples first (scale to the smallest library size) according to Bolaños et al 2021.

A strong reason to subsample is to standardize effort. The bottom line is that in all experimental design you should not be comparing things to which you devote different effort in resolution. For instance, you don't sample one site once a week and another once a month if you want to compare the dynamics between the sites. You standardize effort.


```{r}

ps_min <-  rarefy_even_depth(sub_ps, sample.size = smin, rngseed = 532898)

```

We can also subset the N2 and the N2S4 data here


```{r}
n2s4_core <- ps_min %>% 
  subset_samples(Cruise == "AT34" & Station == 4 & eddy == "Core")

n3s6 <- ps_min %>% 
  subset_samples(Cruise == "AT38" & Station == 6)

s4s6_core <- ps_min %>% 
  subset_samples(Cruise == "AT38" & Station == 6 | Cruise == "AT34" & Station == 4 & eddy == "Core")

s4_core_s6_mz <- s4s6_core %>% 
  subset_samples(Cruise != "AT38" | Station != 6 | z_interv != "Euphotic")
```

## Unconstrained Ordination

One of the best exploratory analyses for amplicon data is unconstrained ordinations. Here we will look at ordinations of our subsampled dataset

### NMDS

Let’s try an NMDS. For NMDS plots it’s important to set a seed since the starting positions of samples in the alogrithm is random.


```{r}
set.seed(7)
# Ordinate
nmds_n2s4_core <- ordinate(n2s4_core, method = "NMDS",  distance = "bray") # stress = 0.10
```

```{r}
set.seed(8)
# Ordinate
nmds_n3s6 <- ordinate(n3s6, method = "NMDS",  distance = "bray") # stress = 0.03
```


```{r}
set.seed(8)
# Ordinate
nmds_s4s6_core <- ordinate(s4s6_core, method = "NMDS",  distance = "bray") # stress = 0.02
```


NMDS plots attempt to show ordinal distances between samples as accurately as possible in two dimensions. It is important to report the stress of these plots, because a high stress value means that the algorithm had a hard time representing the distances between samples in 2 dimensions. The stress of these plots were good - it was .1 (generally anything below .2 is considered acceptable).


```{r echo = FALSE, warning = FALSE, message = FALSE, fig.height = 8, fig.width = 10, fig.align = "center"}

nmds_s4s6_core.plot <- plot_ordination(s4s6_core, nmds_s4s6_core,  title = "NAAMES 2 Station 4 & NAAMES 3 Station 6") +
  # stat_ellipse(aes(color = Cruise2, group = interaction(Cruise2)), type = "t") +
  stat_ellipse(aes(linetype = z_interv, group = interaction(Cruise2, z_interv)), type = "t") +
  geom_point(aes(fill = Cruise2, shape = z_interv), alpha = 0.7, stroke = 2, size = 4) +
  scale_shape_manual(values = c(21, 22)) +
  scale_color_manual(values = custom.colors) +
  scale_fill_manual(values = custom.colors) +
  theme_classic2(base_size = 16)
#removing one of the plotting layers (there are points within points)
nmds_s4s6_core.plot$layers <- nmds_s4s6_core.plot$layers[-1]
```


```{r echo = FALSE, warning = FALSE, message = FALSE, fig.height = 8, fig.width = 10, fig.align = "center"}

nmds_n2s4_core.plot <- plot_ordination(n2s4_core, nmds_n2s4_core,  title = "NAAMES 2 Station 4") +
  stat_ellipse(aes(group = interaction(days), linetype = factor(days)), type = "t") +
  # stat_ellipse(aes(linetype = z_interv, group = interaction(z_interv)), type = "t") +
  # geom_polygon(aes(fill = days, group = interaction(days)), alpha = 0.5) +
  geom_point(aes(fill = days, shape = z_interv), alpha = 0.7, stroke = 2, size = 4) +
  scale_shape_manual(values = c(21, 22, 23)) +
  # scale_color_gradientn(colors = rev(matlab.colors2)) +
  # scale_fill_gradientn(colors = rev(matlab.colors2)) +
  scale_fill_viridis() +
  theme_classic2(base_size = 16)
#removing one of the plotting layers (there are points within points)
nmds_n2s4_core.plot$layers <- nmds_n2s4_core.plot$layers[-1]

```

```{r echo = FALSE, warning = FALSE, message = FALSE, fig.height = 8, fig.width = 10, fig.align = "center"}

nmds_n3s6.plot <- plot_ordination(n3s6, nmds_n3s6,  title = "NAAMES 3 Station 6") +
  stat_ellipse(aes(linetype = z_interv, group = interaction(z_interv)), type = "t") +
   # stat_ellipse(aes(linetype = factor(days), group = interaction(z_interv, days)), type = "t") +
  # geom_polygon(aes(fill = days, group = interaction(days, z_interv)), alpha = 0.5) +
  geom_point(aes(fill = days, shape = z_interv), alpha = 0.7, stroke = 2, size = 4) +
  scale_shape_manual(values = c(21, 22, 23)) +
  # scale_color_gradientn(colors = rev(matlab.colors2)) +
  # scale_fill_gradientn(colors = rev(matlab.colors2)) +
  scale_fill_viridis() +
  theme_classic2(base_size = 16)
#removing one of the plotting layers (there are points within points)
nmds_n3s6.plot$layers <- nmds_n3s6.plot$layers[-1]

```



```{r fig.height=6, fig.width=23}

(nmds_s4s6_core.plot + 
  guides( shape = guide_legend(title = "Depth Horizon"), color = F,  linetype = guide_legend(title = "Depth Horizon"), fill = guide_legend(title = "Cruise", override.aes = list(shape = 21)))) +
  

(nmds_n2s4_core.plot + 
  guides(fill = guide_colorbar(title = "Days", reverse = T), shape = guide_legend(title = "Depth Horizon"), linetype = guide_legend(title = "Days"), color = F)) +
  
  (nmds_n3s6.plot + 
  guides(fill = guide_colorbar(title = "Days", reverse = T), shape = guide_legend(title = "Depth Horizon"),  linetype = guide_legend(title = "Depth Horizon"), color = F)) +
  

  plot_annotation(tag_levels = "a") &
  theme(plot.tag = element_text(size = 22),
        plot.title = element_text(size = 18)) 

```

95% confidence interval ellipse for the mean (group centroid) tells us something about the sampling distribution of the mean (centroid) we might see if we repeated your data collection a lot of times. In other words we are looking at the uncertainty in the estimate of the population mean (centroid) given the sample of data we collected.


# Betadisper and permutational ANOVA

Above, we performed beta diversity analyses on Bray-Curtis distances on
rarefied datasets that were then visualized using NMDS.
We can test if there are statistically significant differences between
sample groups using the betadisper and adonis functions of the vegan
package. Betadisper tests whether two or more groups are homogeneously
dispersed in relation to their species in studied samples. This test can be done to see if one group has more compositional variance than
another. Moreover, homogeneity of dispersion among groups is very
advisable to have if you want to test if two or more groups have
different compositions, which is tested by adonis.

## Phyloseq to DESeq, distance matrix

To be able to run the stats, we first have to create a distance matrix
from our data. We’ll use the DESeq package to do so.

```{r message=FALSE, warning=FALSE}
library(DESeq2)
library(vegan)
```

```{r}
deseq_counts <- phyloseq_to_deseq2(ps_min, design = ~latitude ) #the design argument is required but doesn't matter here
```

```{r}
deseq_count_tab <- assay(deseq_counts) #extract the read count matrix
```

We’ll calculate bray-curtis distances, which reflect the NMDS ordinations above


```{r}
#We can subset our data if we want to and calculate distances/run stats for only a subset of the group. The code below shows how

sample.tab2 <- sample.tab %>%
  rownames_to_column() %>% 
  filter(z %in% c(5, 25, 50, 75, 100, 150, 200)) %>% 
  column_to_rownames(var = "rowname")


subset_sample_IDs_n2s4_ez <-  row.names(sample.tab2)[sample.tab2$Station == 4 & sample.tab2$Cruise == "AT34" & sample.tab2$eddy == "Core" & sample.tab2$z <= 75]

dist_n2s4_ez <- vegdist(t(deseq_count_tab[ , colnames(deseq_count_tab) %in% subset_sample_IDs_n2s4_ez]), method = "bray")

sample_info_tab_n2s4_ez <- sample.tab2[row.names(sample.tab2) %in% subset_sample_IDs_n2s4_ez, ]

####


subset_sample_IDs_n2s4_mz <-  row.names(sample.tab2)[sample.tab2$Station == 4 & sample.tab2$Cruise == "AT34" & sample.tab2$eddy == "Core" & sample.tab2$z >= 100]

dist_n2s4_mz <- vegdist(t(deseq_count_tab[ , colnames(deseq_count_tab) %in% subset_sample_IDs_n2s4_mz]), method = "bray")

sample_info_tab_n2s4_mz <- sample.tab2[row.names(sample.tab2) %in% subset_sample_IDs_n2s4_mz, ]


####

subset_sample_IDs_n3s6 <-  row.names(sample.tab2)[sample.tab2$Station == 6 & sample.tab2$Cruise == "AT38"]

dist_n3s6 <- vegdist(t(deseq_count_tab[ , colnames(deseq_count_tab) %in% subset_sample_IDs_n3s6]), method = "bray")

sample_info_tab_n3s6 <- sample.tab2[row.names(sample.tab2) %in% subset_sample_IDs_n3s6, ]

####

subset_sample_IDs_n3s6_ez <-  row.names(sample.tab2)[sample.tab2$Station == 6 & sample.tab2$Cruise == "AT38" & sample.tab2$z <= 75]

dist_n3s6_ez <- vegdist(t(deseq_count_tab[ , colnames(deseq_count_tab) %in% subset_sample_IDs_n3s6_ez]), method = "bray")

sample_info_tab_n3s6_ez <- sample.tab2[row.names(sample.tab2) %in% subset_sample_IDs_n3s6_ez, ]

####

subset_sample_IDs_n3s6_mz <-  row.names(sample.tab2)[sample.tab2$Station == 6 & sample.tab2$Cruise == "AT38" & sample.tab2$z >= 100]

dist_n3s6_mz <- vegdist(t(deseq_count_tab[ , colnames(deseq_count_tab) %in% subset_sample_IDs_n3s6_mz]), method = "bray")

sample_info_tab_n3s6_mz <- sample.tab2[row.names(sample.tab2) %in% subset_sample_IDs_n3s6_mz, ]


```

Betadisper first calculates the average distance of group members to the group centroid in multivariate space (generated by a distance matrix).

In the function below: we are using the distance matrix to calculate the multivariate dispersions (variances; average distance to centroids). We then use group dispersions to perform an ANOVA test.



```{r}
anova(betadisper(dist_n2s4_ez, sample_info_tab_n2s4_ez$days)) 
```

The ANOVA’s p-value is not significant meaning that group dispersions are homogenous (“Null hypothesis of no difference in dispersion between groups”)

```{r}
anova(betadisper(dist_n2s4_mz, sample_info_tab_n2s4_mz$days)) 
```


```{r}
anova(betadisper(dist_n3s6, sample_info_tab_n3s6$z_interv)) 
```

```{r}
anova(betadisper(dist_n3s6_ez, sample_info_tab_n3s6_ez$days)) 
```
```{r}
anova(betadisper(dist_n3s6_mz, sample_info_tab_n3s6_mz$days)) 
```

**Our groups (depth horizon, Days) within N2S4 and N3S6 do present homogeneity among group dispersions (compositions vary similarly). The exception is euphotic zone N3S6 (p = 0.02)** 

```{r}
adonis(dist_n2s4_ez~sample_info_tab_n2s4_ez$days)
```
```{r}
adonis(dist_n2s4_mz~sample_info_tab_n2s4_mz$days)
```

```{r}
adonis(dist_n3s6~sample_info_tab_n3s6$z_interv)
```


```{r}
adonis(dist_n3s6_ez~sample_info_tab_n3s6_ez$days)
```

```{r}
adonis(dist_n3s6_mz~sample_info_tab_n3s6_mz$days)
```

Adonis analyzes and partitions sums of squares using distance matrices. It can be seen as an ANOVA using distance matrices (analogous to MANOVA – multivariate analysis of variance). Therefore, it is used to test if two or more groups have similar compositions.

**Our groups (depth horizon, days) at N2S4 present homogeneity among group dispersions (compositions vary similarly). Only in the mesopelagic do communities shift **

**Our groups (depth horizon, days) at N3S6 Ez present homogeneity among group dispersions (compositions vary similarly) and do not have significantly different compositions. Our groups at N3S6 Ez don not present homogeneity among group dispersions**


# Alpha Diversity

We are going to calculate the the Chao1 diversity index. 

**it is important to note that the alpha diversity values are not interpretable as “real” numbers of anything (due to the nature of amplicon data), but they can still be useful as relative metrics of comparison.**

[Chao1: nonparametric estimation of minimum community richness](https://www.jstor.org/stable/4615964?seq=1#metadata_info_tab_contents) 


```{r}
richness_n2s4 <- estimate_richness(n2s4_core, measures = c("Chao1", "Shannon")) %>% 
  rownames_to_column(., var = "DNA_ID") %>% 
  mutate_at(vars(DNA_ID), str_replace_all, pattern = "NAAMES2.", "NAAMES2-") 
```

```{r}
richness_n3s6 <- estimate_richness(n3s6, measures = c("Chao1", "Shannon")) %>% 
  rownames_to_column(., var = "DNA_ID") %>% 
  mutate_at(vars(DNA_ID), str_replace_all, pattern = "N3S6C4.", "N3S6C4-") %>% 
  mutate_at(vars(DNA_ID), str_replace_all, pattern = "N3S6C5.", "N3S6C5-") %>% 
  mutate_at(vars(DNA_ID), str_replace_all, pattern = "N3S6C7.", "N3S6C7-") %>% 
  mutate_at(vars(DNA_ID), str_replace_all, pattern = "N3S6C10.", "N3S6C10-")  
```


```{r}
richness_s4_core_s6_mz <- estimate_richness(s4_core_s6_mz, measures = c("Chao1", "Shannon")) %>% 
  rownames_to_column(., var = "DNA_ID") %>% 
  mutate_at(vars(DNA_ID), str_replace_all, pattern = "NAAMES2.", "NAAMES2-") %>% 
  mutate_at(vars(DNA_ID), str_replace_all, pattern = "N3S6C4.", "N3S6C4-") %>% 
  mutate_at(vars(DNA_ID), str_replace_all, pattern = "N3S6C5.", "N3S6C5-") %>% 
  mutate_at(vars(DNA_ID), str_replace_all, pattern = "N3S6C7.", "N3S6C7-") %>% 
  mutate_at(vars(DNA_ID), str_replace_all, pattern = "N3S6C10.", "N3S6C10-")  
```



Let’s add the sample metadata into these dataframes 


```{r}
alphadiv_n2s4 <- left_join(richness_n2s4, sample.tab2 %>% rownames_to_column(., var = "DNA_ID")) 
```
```{r}
alphadiv_n3s6 <- left_join(richness_n3s6, sample.tab2 %>% rownames_to_column(., var = "DNA_ID")) 
```

```{r}
alphadiv_s4_core_s6_mz <- left_join(richness_s4_core_s6_mz, sample.tab2 %>% rownames_to_column(., var = "DNA_ID"))  
```



```{r echo = FALSE, warning = FALSE, message = FALSE, fig.height = 6, fig.width = 8, fig.align = "center"}

my_comparisons <- list( c("0", "1"), c("0", "2"), c("0", "3"), c("1", "2"), c("1", "3"), c("2", "3")) 


 alpha.plot_n2s4 <- alphadiv_n2s4 %>% 
   ggplot(aes(x = factor(z_interv), y = Chao1)) +
   geom_boxplot(width = 0.3, outlier.shape = NA) +
   geom_point(aes(fill = z), shape = 21, size = 3, alpha = 0.7) +
  scale_color_viridis_c(option = "viridis", direction = -1) +
  scale_fill_viridis_c(option = "viridis", direction = -1) +
  theme_classic2(base_size = 16) +
  labs(x = "", y =  "Chao1", fill = "Depth, m") + 
  stat_compare_means(method = "t.test")  +
  # stat_compare_means(comparisons = my_comparisons,
  #                    label = "p.adj",
  #                    step.increase = 0.11,
  #                    tip.length = 0.01) +
  facet_grid(~ factor(days), scales = "free") +
   ylim(250, 850) +
   ggtitle("NAAMES 2 Station 4 (N2S4)") +
   guides(colour = guide_colorbar(reverse = T),
         fill = guide_colorbar(reverse = T))
```

```{r echo = FALSE, warning = FALSE, message = FALSE, fig.height = 6, fig.width = 8, fig.align = "center"}

my_comparisons <- list( c("0", "1"), c("0", "2"),  c("1", "2")) 


alpha.plot_n3s6 <- alphadiv_n3s6 %>% 
   ggplot(aes(x = factor(z_interv), y = Chao1)) +
   geom_boxplot(width = 0.3, outlier.shape = NA) +
   geom_point(aes(fill = z), shape = 21, size = 3, alpha = 0.7) +
  scale_color_viridis_c(option = "viridis", direction = -1) +
  scale_fill_viridis_c(option = "viridis", direction = -1) +
  theme_classic2(base_size = 16) +
  labs(x = "", y =  "Chao1", fill = "Depth, m") + 
  stat_compare_means(method = "t.test")  +
  # stat_compare_means(comparisons = my_comparisons,
  #                    label = "p.adj",
  #                    step.increase = 0.11,
  #                    tip.length = 0.01) +
  facet_grid(~ factor(days), scales = "free") +
  ylim(250, 850) +
  ggtitle("NAAMES 3 Station 6 (N3S6)") +
  guides(fill = F)
```

```{r echo = FALSE, warning = FALSE, message = FALSE, fig.height = 6, fig.width = 8, fig.align = "center"}

my_comparisons <- list( c("0", "1"), c("0", "2"),  c("1", "2")) 


alpha.plot_s4_core_s6_mz <- alphadiv_s4_core_s6_mz %>% 
   ggplot(aes(x = Cruise2, y = Chao1)) +
   geom_boxplot(width = 0.3, outlier.shape = NA) +
   geom_point(aes(fill = factor(z_interv)), shape = 21, size = 3, alpha = 0.7) +
   scale_color_manual(values = custom.colors) +
  scale_fill_manual(values = custom.colors) +
  theme_classic2(base_size = 16) +
  labs(x = "", y =  "Chao1", fill = "Depth Horizon") + 
  stat_compare_means(method = "t.test")  +
  # stat_compare_means(comparisons = my_comparisons,
  #                    label = "p.adj",
  #                    step.increase = 0.11,
  #                    tip.length = 0.01) +
  facet_grid(~ factor(days), scales = "free") +
  ylim(250, 850) +
  ggtitle("N2S4 & Upper Mesopelagic N3S6") +
  guides()
```


```{r fig.height=12, fig.width=12}
alpha.plot_n2s4 / alpha.plot_n3s6 / alpha.plot_s4_core_s6_mz + 
  
  plot_layout(guides = 'collect') +
  plot_annotation(tag_levels = "a") &
  theme(plot.tag = element_text(size = 22),
        plot.title = element_text(size = 18)) 

```


Boxes represent the 1.5 interquartile range, with the internal solid line representing the median. Circles represent data points. Difference in the alpha diversity indexes among conditions were tested using pairwise t-tests; p < 0.05 was considered the threshold significance for a difference between conditions.


# Who??

Which taxa were important? Which taxa were contributing to differences in community compositon?

**Note: Recovered 16S rRNA gene copy numbers do not equal organism abundance.**

That said, we can generate a heat map of our samples showing us how the relative abundance of different taxonomic groups change...potentially giving us a visual of which taxa are most important to the alpha and beta diversity patterns we observed. 
First, we're going to generate a custom table that will be easier to work with than a phyloseq object.



## Generate relative abundances

Our data currently shows number gene copies recovered, so we'll convert to percentages (relative abundances)

```{r}
# n2s4_core <- n2s4 %>% 
#   subset_samples(eddy == "Core")
ps_std <- transform_sample_counts(n2s4_core, function(x) x/sum(x))
#extract the relative abundance table and coerce into dataframe
ps_std.tab <- as(otu_table(ps_std), "matrix")
ps_std.df = as.data.frame(ps_std.tab) 
```

```{r}
ps_std_s6 <- transform_sample_counts(n3s6, function(x) x/sum(x))
#extract the relative abundance table and coerce into dataframe
ps_std.tab_s6 <- as(otu_table(ps_std_s6), "matrix")
ps_std.df_s6 = as.data.frame(ps_std.tab_s6) 
```

### Make table

```{r warning = F}
#first coerce the taxa table into a data frame
tax.df <-  as.data.frame(tax.tab) 
#then combine the data frames
custom.tab <- tax.df %>% 
  rownames_to_column(., var = "asv") %>% 
  left_join(., ps_std.df %>% rownames_to_column(., var = "asv")) %>% 
  #create a new index of that combines the  class, order, family, and genus values, you can play around here!!
  mutate(#pcofg = paste(Phylum, "_", Class, "_", Order,"_", Family, "_", Genus),
         pcof = paste(Phylum, "_", Class, "_", Order,"_", Family)) %>% 
         # pco = paste(Phylum, "_", Class, "_", Order)) %>% 
  select(-c(asv:Genus)) %>% 
  select(pcof,everything()) %>%
  group_by(pcof) %>%
  # select(pco,everything()) %>% 
  # group_by(pco) %>% 
  #here we are combining the relative abundances based on our grouping
  summarise_at(vars(!contains(c("pco"))), sum, na.rm = T) %>% 
  ungroup()

#save the row names and then make them into the column names
colnames <- custom.tab[,1] 

#transpose the dataframe so we can merge with the sample info table
t_custom.tab <-  as.data.frame(t(custom.tab[,-1]))
colnames(t_custom.tab) <- colnames$pcof
# colnames(t_custom.tab) <- colnames$pco

#merge
sweet.tab <- t_custom.tab %>% 
  rownames_to_column(., var = "sample") %>% 
  left_join(., sample.tab %>% rownames_to_column(., var = "sample")) %>% 
  select(sample, Cruise:npp, everything())


relabund <- sweet.tab %>% 
  select(-c(sample:npp)) %>% 
  #remove groups that are completely absent
  .[ , colSums(.) > 0] %>% 
  #arrange by biggest contributors
  .[, order(colSums(-.))] %>% 
  bind_cols(sweet.tab %>% select(sample:npp), .)
```

```{r warning = F}
#then combine the data frames
custom.tab_s6 <- tax.df %>% 
  rownames_to_column(., var = "asv") %>% 
  left_join(., ps_std.df_s6 %>% rownames_to_column(., var = "asv")) %>% 
  #create a new index of that combines the  class, order, family, and genus values, you can play around here!!
  mutate(#pcofg = paste(Phylum, "_", Class, "_", Order,"_", Family, "_", Genus),
         pcof = paste(Phylum, "_", Class, "_", Order,"_", Family)) %>% 
         # pco = paste(Phylum, "_", Class, "_", Order)) %>% 
  select(-c(asv:Genus)) %>% 
  select(pcof,everything()) %>%
  group_by(pcof) %>%
  # select(pco,everything()) %>% 
  # group_by(pco) %>% 
  #here we are combining the relative abundances based on our grouping
  summarise_at(vars(!contains(c("pco"))), sum, na.rm = T) %>% 
  ungroup()

#save the row names and then make them into the column names
colnames_s6 <- custom.tab_s6[,1] 

#transpose the dataframe so we can merge with the sample info table
t_custom.tab_s6 <-  as.data.frame(t(custom.tab_s6[,-1]))
colnames(t_custom.tab_s6) <- colnames_s6$pcof
# colnames(t_custom.tab) <- colnames$pco

#merge
sweet.tab_s6 <- t_custom.tab_s6 %>% 
  rownames_to_column(., var = "sample") %>% 
  left_join(., sample.tab %>% rownames_to_column(., var = "sample")) %>% 
  select(sample, Cruise:npp, everything())


relabund_s6 <- sweet.tab_s6 %>% 
  select(-c(sample:npp)) %>% 
  #remove groups that are completely absent
  .[ , colSums(.) > 0] %>% 
  #arrange by biggest contributors
  .[, order(colSums(-.))] %>% 
  bind_cols(sweet.tab_s6 %>% select(sample:npp), .)
```


## Heatmaps

```{r echo = FALSE, warning = FALSE, message = FALSE, fig.height = 8, fig.width = 12, fig.align = "center"}
relaheat.data <- relabund %>% 
  select(-c(sample, Cruise:CampCN, mld:npp)) %>%
  pivot_longer(.,-c(eddy, days, z, z_interv), names_to = "taxa", values_to = "relabund") %>% 
  separate(taxa, into = c("p", "c", "o", "f"), sep = " _ ") %>% 
  mutate(cof = paste(c, o, f, sep = ": ")) %>% 
  group_by(days, z, cof) %>% 
  mutate(mean_relabund = mean(relabund, na.rm = T)) %>% 
  ungroup() %>% 
  select(-relabund) %>% 
  distinct() %>% 
  mutate_at(vars(p,c,o, cof), str_replace_all, "NA", "Unassigned") %>% 
  arrange(desc(cof), mean_relabund)

# install.packages("viridis")
# library(viridis)

relaheat_cof <- relaheat.data %>%
  filter(mean_relabund > 0.001) %>%
  ggplot(aes(x = days, y = reorder(cof, mean_relabund))) +
  geom_tile(aes(fill = mean_relabund), color = "white") +
  # scale_color_manual(values = custom.colors) %>%
  # scale_fill_gradientn(colors = rev(matlab.colors)) +
  scale_fill_viridis(option = "D", begin = 0.01) +
  labs(x = "Days", y = "Class: Order: Family", fill = "Relative Abundance") +
  facet_grid(~z) +
  theme_classic2(base_size = 16) +
  theme(axis.text.y = element_text(size = 12), legend.position = "top") +
   guides(fill = guide_colourbar(barheight = 2, barwidth = 20, frame.colour = "black", frame.linewidth = 2,ticks.colour = "black", ticks.linewidth = 1), color = F) +
  ggtitle("NAAMES 2 Station 4")


```



```{r echo = FALSE, warning = FALSE, message = FALSE, fig.height = 8, fig.width = 12, fig.align = "center"}
relaheat.data_s6 <- relabund_s6 %>% 
  select(-c(sample, Cruise:CampCN, mld:npp)) %>%
  pivot_longer(.,-c(eddy,  days, z, z_interv), names_to = "taxa", values_to = "relabund") %>% 
  separate(taxa, into = c("p", "c", "o", "f"), sep = " _ ") %>% 
  mutate(cof = paste(c, o, f, sep = ": ")) %>% 
  group_by(days, z, cof) %>% 
  mutate(mean_relabund = mean(relabund, na.rm = T)) %>% 
  ungroup() %>% 
  select(-relabund) %>% 
  distinct() %>% 
  mutate_at(vars(p,c,o, cof), str_replace_all, "NA", "Unassigned") %>% 
  arrange(desc(cof), mean_relabund) 

# install.packages("viridis")
# library(viridis)

relaheat_cof_s6 <- relaheat.data_s6 %>%
  filter(mean_relabund > 0.001) %>%
  ggplot(aes(x = days, y = reorder(cof, mean_relabund))) +
  geom_tile(aes(fill = mean_relabund), color = "white") +
  # scale_color_manual(values = custom.colors) %>%
  # scale_fill_gradientn(colors = rev(matlab.colors)) +
  scale_fill_viridis(option = "D", begin = 0.01) +
  # scale_fill_viridis(option = "D", begin = 0.01) +
  labs(x = "Days", y = "Class: Order: Family", fill = "Relative Abundance") +
  facet_grid(~z) +
  theme_classic2(base_size = 16) +
  theme(axis.text.y = element_text(size = 12), legend.position = "top") +
   guides(fill = guide_colourbar(barheight = 2, barwidth = 20, frame.colour = "black", frame.linewidth = 2,ticks.colour = "black", ticks.linewidth = 1), color = F) +
  ggtitle("NAAMES 3 Station 6")


```


## Combined counts/relative abundnace

here we combine the counts for our two main deph horizons, 5 -75 m and 100 - 200 m

### Make table

```{r warning = F}

combine_s4 <- n2s4_core %>% 
  transform_sample_counts(., function(x) x*1) #not actually transforming counts here...just converting frpm phyloseq object
 combine_s4.df <-  as(otu_table(combine_s4), "matrix") %>%
   as.data.frame(.) 


#then combine the data frames
custom.tab_combine_s4 <- tax.df %>% 
  rownames_to_column(., var = "asv") %>% 
  left_join(., combine_s4.df %>% rownames_to_column(., var = "asv")) %>% 
  #create a new index of that combines the  class, order, family, and genus values, you can play around here!!
  mutate(#pcofg = paste(Phylum, "_", Class, "_", Order,"_", Family, "_", Genus),
         pcof = paste(Phylum, "_", Class, "_", Order,"_", Family)) %>% 
         # pco = paste(Phylum, "_", Class, "_", Order)) %>% 
  select(-c(asv:Genus)) %>% 
  select(pcof,everything()) %>%
  group_by(pcof) %>%
  # select(pco,everything()) %>% 
  # group_by(pco) %>% 
  #here we are combining the relative abundances based on our grouping
  summarise_at(vars(!contains(c("pcof"))), sum, na.rm = T) %>% 
  ungroup()

#save the row names and then make them into the column names
colnames <- custom.tab_combine_s4[,1] 

#transpose the dataframe so we can merge with the sample info table
t_custom.tab_combine_s4 <-  as.data.frame(t(custom.tab_combine_s4[,-1]))
colnames(t_custom.tab_combine_s4) <- colnames$pcof
# colnames(t_custom.tab) <- colnames$pco

#merge
sweet.tab_custom.tab_combine_s4  <- t_custom.tab_combine_s4  %>% 
  rownames_to_column(., var = "sample") %>% 
  left_join(., sample.tab %>% rownames_to_column(., var = "sample")) %>% 
  select(sample, Cruise:npp, everything())


counts_custom.tab_combine_s4  <- sweet.tab_custom.tab_combine_s4 %>% 
  select(-c(sample:npp)) %>% 
  #remove groups that are completely absent
  .[ , colSums(.) > 0] %>% 
  #arrange by biggest contributors
  .[, order(colSums(-.))] %>% 
  bind_cols(sweet.tab_custom.tab_combine_s4 %>% select(sample:npp), .) %>% 
  mutate_at(vars(days), as.character) %>% 
  group_by(days, z_interv) %>% 
  summarise_at(vars(!c(sample:npp)), sum, na.rm = T) %>% 
  ungroup() %>% 
  rowwise() %>% 
  mutate(total = sum(c_across(where(is.numeric)))) %>% 
  ungroup() %>% 
  select(days:z_interv, total, everything())  %>% 
  mutate(across(c(4:140), ~ . / total)) %>% 
  select(-total)
  
```


### Make table at order level

```{r warning = F}

#then combine the data frames
custom.tab_combine_s4_order <- tax.df %>% 
  rownames_to_column(., var = "asv") %>% 
  left_join(., combine_s4.df %>% rownames_to_column(., var = "asv")) %>% 
  #create a new index of that combines the  class, order, family, and genus values, you can play around here!!
  mutate( pco = paste(Phylum, "_", Class, "_", Order)) %>% 
  select(-c(asv:Genus)) %>% 
  select(pco,everything()) %>%
  group_by(pco) %>%
  #here we are combining the relative abundances based on our grouping
  summarise_at(vars(!contains(c("pco"))), sum, na.rm = T) %>% 
  ungroup()

#save the row names and then make them into the column names
colnames <- custom.tab_combine_s4_order[,1] 

#transpose the dataframe so we can merge with the sample info table
t_custom.tab_combine_s4_order <-  as.data.frame(t(custom.tab_combine_s4_order[,-1]))
colnames(t_custom.tab_combine_s4_order) <- colnames$pco
# colnames(t_custom.tab) <- colnames$pco

#merge
sweet.tab_custom.tab_combine_s4_order  <- t_custom.tab_combine_s4_order  %>% 
  rownames_to_column(., var = "sample") %>% 
  left_join(., sample.tab %>% rownames_to_column(., var = "sample")) %>% 
  select(sample, Cruise:npp, everything())


counts_custom.tab_combine_s4_order  <- sweet.tab_custom.tab_combine_s4_order %>% 
  select(-c(sample:npp)) %>% 
  #remove groups that are completely absent
  .[ , colSums(.) > 0] %>% 
  #arrange by biggest contributors
  .[, order(colSums(-.))] %>% 
  bind_cols(sweet.tab_custom.tab_combine_s4_order %>% select(sample:npp), .) %>% 
  mutate_at(vars(days), as.character) %>% 
  group_by(days, z_interv) %>% 
  summarise_at(vars(!c(sample:npp)), sum, na.rm = T) %>% 
  ungroup() %>% 
  rowwise() %>% 
  mutate(total = sum(c_across(where(is.numeric)))) %>% 
  ungroup() %>% 
  select(days:z_interv, total, everything())  %>% 
  mutate(across(c(4:87), ~ . / total)) %>% 
  select(-total)
  
```


## Heatmaps

```{r echo = FALSE, warning = FALSE, message = FALSE, fig.height = 8, fig.width = 12, fig.align = "center"}
relaheat.data_combine_s4 <- counts_custom.tab_combine_s4 %>% 
  pivot_longer(.,-c(days:z_interv), names_to = "taxa", values_to = "relabund") %>% 
  separate(taxa, into = c("p", "c", "o", "f"), sep = " _ ") %>% 
  mutate(pcof = paste(p, c, o, f, sep = ": ")) %>% 
  group_by(days, z_interv, pcof) %>% 
  mutate(mean_relabund = mean(relabund, na.rm = T),
         sd_relabund = sd(relabund, na.rm = T)) %>% 
  ungroup() %>% 
  select(-relabund) %>% 
  distinct() %>% 
  mutate_at(vars(p,c,o, pcof), str_replace_all, "NA", "Unassigned") %>% 
  arrange(desc(pcof), mean_relabund)

rela.counts <-  read_rds("~/GITHUB/naames_multiday/Output/processed_data.rds") %>% 
  filter(Cruise == "AT34" & Station == 4) %>% 
  mutate_at(vars(contains("tdaa"), Asp:Lys), funs(. / 10^3)) %>% #nM to mmol/m3
  mutate(time = ymd_hms(datetime),
         interv = interval(dplyr::first(time), time),
         dur = as.duration(interv),
         days = as.numeric(dur, "days"),
         eddy = ifelse(Date != "2016-05-27", "Core", "Periphery"),
         tdaa_yield = round((tdaa_c/doc)*100, 2)) %>% 
  filter(z <= 200) %>% 
  filter(eddy == "Core") %>% 
  select(days, z, ba) %>% 
  drop_na(ba) %>% 
  mutate(days = round(days)) %>% 
  left_join(relaheat.data, .) %>% 
  select(eddy:z_interv, ba) %>% 
  distinct() %>% 
  group_by(days, z_interv) %>% 
  mutate(sum_ba = sum(ba)) %>% 
  ungroup() %>% 
  select(days, z_interv, sum_ba) %>% 
  distinct() %>% 
  arrange(days) %>% 
  mutate_at(vars(days), as.character) %>% 
  left_join(., relaheat.data_combine_s4) %>% 
  mutate(relacount = mean_relabund * sum_ba) %>% 
  left_join(., gene_copies %>% 
  dplyr::rename(p = Phylum,
                c = Class, 
                o = Order) %>% select(p, c, o, copy_num)) %>% 
  mutate(norm_count = relacount/copy_num)


relaheat_combine_s4 <- relaheat.data_combine_s4 %>%
  filter(mean_relabund > 0.001) %>%
  select(days, z_interv, pcof, mean_relabund) %>% 
  distinct() %>% 
  ggplot(aes(x = days, y = reorder(pcof, mean_relabund))) +
  geom_tile(aes(fill = mean_relabund), color = "white") +
  # scale_color_manual(values = custom.colors) %>%
  # scale_fill_gradientn(colors = rev(matlab.colors)) +
  scale_fill_viridis_b(option = "D",  trans = 'log10') +
  geom_text(aes(label = round(mean_relabund, 2), color = "white"), size = 4) +
  scale_color_manual(values = c("white" = "white", "black" = "black")) +
  labs(x = "Days", y = "Phylum: Class: Order: Family", fill = "Relative Abundance") +
  facet_grid(~factor(z_interv, levels = levels)) +
  theme_classic2(base_size = 16) +
  theme(axis.text.y = element_text(size = 12), legend.position = "top") +
   guides(fill = guide_colourbar(barheight = 2, barwidth = 20, frame.colour = "black", frame.linewidth = 2,ticks.colour = "black", ticks.linewidth = 1), color = F) +
  ggtitle("")


```

```{r fig.height=8, fig.width=12}
relacounts_combine_s4 <- rela.counts %>%
  filter(mean_relabund > 0.001) %>%
  select(days, z_interv, pcof, mean_relabund, relacount, norm_count) %>% 
  distinct() %>% 
  ggplot(aes(x = days, y = reorder(pcof, mean_relabund))) +
  geom_tile(aes(fill = norm_count), color = "white") +
  # scale_color_manual(values = custom.colors) %>%
  # scale_fill_gradientn(colors = rev(matlab.colors)) +
  scale_fill_viridis_b(option = "D",  trans = 'log10') +
  geom_text(aes(label = formatC(norm_count, format = "e", digits = 1), color = "white"), size = 4) +
  scale_color_manual(values = c("white" = "white", "black" = "black")) +
  labs(x = "Days", y = "Phylum: Class: Order: Family", fill = expression(paste("Cells L"^-1))) +
  facet_grid(~factor(z_interv, levels = levels)) +
  theme_classic2(base_size = 16) +
  theme(axis.text.y = element_text(size = 12), legend.position = "top") +
   guides(fill = guide_colourbar(barheight = 2, barwidth = 20, frame.colour = "black", frame.linewidth = 2,ticks.colour = "black", ticks.linewidth = 1), color = F) +
  ggtitle("") 
```


```{r}
del_data_combine_s4 <- rela.counts  %>% 
  group_by(z_interv, pcof) %>% 
  mutate(init_relabund = ifelse(days == 0, mean_relabund, NA),
         init_count = ifelse(days == 0, relacount, NA)) %>% 
  fill(c(init_relabund, init_count), .direction = "updown") %>% 
  mutate(del_relabund = ifelse(days > 0, mean_relabund - init_relabund, 0),
         del_count = ifelse(days > 0, relacount - init_count, 0)) %>% 
  ungroup()
```



```{r fig.height=8, fig.width=12}
relaheat_del_combine_s4 <- del_data_combine_s4  %>%
  select(days, z_interv, pcof, mean_relabund, del_relabund) %>% 
  distinct() %>% 
  filter(mean_relabund > 0.001, days > 0) %>%
  ggplot(aes(x = days, y = reorder(pcof, mean_relabund))) +
  geom_tile(aes(fill = round(del_relabund, 2)), color = "white") +
  # scale_fill_gradient2(low = "#313695", mid = "#FEE090", high = "#A50026", midpoint = 0) +
  # scale_fill_gradientn(colors = rev(odv.colors)) +
   # scale_fill_viridis_b() +
  scale_fill_viridis_c() +
  geom_text(aes(label = round(del_relabund, 2), color = "white"), size = 4) +
  scale_color_manual(values = c("white" = "white", "black" = "black")) +
  labs(x = "Days", y = "", fill = "∆ Relative Abundance") +
  facet_grid(~factor(z_interv, levels = levels)) +
  theme_classic2(base_size = 16) +
  theme( legend.position = "top", axis.text.y = element_blank()) +
  # theme( legend.position = "top") +
   guides(fill = guide_colourbar(barheight = 2, barwidth = 20, frame.colour = "black", frame.linewidth = 2,ticks.colour = "black", ticks.linewidth = 1), color = F) 

```


```{r fig.height=8, fig.width=12}
relacount_del_combine_s4 <- del_data_combine_s4  %>%
  select(days, z_interv, pcof, mean_relabund, del_relabund, del_count) %>% 
  distinct() %>% 
  filter(mean_relabund > 0.001, days > 0) %>%
  ggplot(aes(x = days, y = reorder(pcof, mean_relabund))) +
  geom_tile(aes(fill = del_count), color = "white") +
  # scale_fill_gradient2(low = "#313695", mid = "#FEE090", high = "#A50026", midpoint = 0) +
  # scale_fill_gradientn(colors = rev(odv.colors)) +
   # scale_fill_viridis_b(begin = 0.2) +
  scale_fill_viridis_c(begin = 0.2) +
  geom_text(aes(label = formatC(del_count, format = "e", digits = 1), color = "white"), size = 4) +
  scale_color_manual(values = c("white" = "white", "black" = "black")) +
  labs(x = "Days", y = "", fill = expression(paste("∆ Cells L"^-1))) +
  facet_grid(~factor(z_interv, levels = levels)) +
  theme_classic2(base_size = 16) +
  theme( legend.position = "top", axis.text.y = element_blank()) +
  # theme( legend.position = "top") +
   guides(fill = guide_colourbar(barheight = 2, barwidth = 20, frame.colour = "black", frame.linewidth = 2,ticks.colour = "black", ticks.linewidth = 1), color = F) 

```


#### heat map order level

```{r echo = FALSE, warning = FALSE, message = FALSE, fig.height = 8, fig.width = 12, fig.align = "center"}
relaheat.data_combine_s4_order <- counts_custom.tab_combine_s4_order %>% 
  pivot_longer(.,-c(days:z_interv), names_to = "taxa", values_to = "relabund") %>% 
  separate(taxa, into = c("p", "c", "o" ), sep = " _ ") %>% 
  mutate(pco = paste(p, c, o, sep = ": ")) %>% 
  group_by(days, z_interv, pco) %>% 
  mutate(mean_relabund = mean(relabund, na.rm = T)) %>% 
  ungroup() %>% 
  select(-relabund) %>% 
  distinct() %>% 
  mutate_at(vars(p,c,o, pco), str_replace_all, "NA", "Unassigned") %>% 
  arrange(desc(pco), mean_relabund)

rela.counts_order <-  read_rds("~/GITHUB/naames_multiday/Output/processed_data.rds") %>% 
  filter(Cruise == "AT34" & Station == 4) %>% 
  mutate_at(vars(contains("tdaa"), Asp:Lys), funs(. / 10^3)) %>% #nM to mmol/m3
  mutate(time = ymd_hms(datetime),
         interv = interval(dplyr::first(time), time),
         dur = as.duration(interv),
         days = as.numeric(dur, "days"),
         eddy = ifelse(Date != "2016-05-27", "Core", "Periphery"),
         tdaa_yield = round((tdaa_c/doc)*100, 2)) %>% 
  filter(z <= 200) %>% 
  filter(eddy == "Core") %>% 
  select(days, z, ba) %>% 
  drop_na(ba) %>% 
  mutate(days = round(days)) %>% 
  left_join(relaheat.data, .) %>% 
  select(eddy:z_interv, ba) %>% 
  distinct() %>% 
  group_by(days, z_interv) %>% 
  mutate(sum_ba = sum(ba)) %>% 
  ungroup() %>% 
  select(days, z_interv, sum_ba) %>% 
  distinct() %>% 
  arrange(days) %>% 
  mutate_at(vars(days), as.character) %>% 
  left_join(., relaheat.data_combine_s4_order) %>% 
  mutate(relacount = mean_relabund * sum_ba) %>% 
  left_join(., gene_copies %>% 
  dplyr::rename(p = Phylum,
                c = Class, 
                o = Order) %>% select(p, c, o, copy_num)) %>% 
  mutate(pco = paste(p, ":", c, ":", o)) %>% 
  mutate(norm_count = relacount/copy_num)


relaheat_combine_s4_order <- rela.counts_order %>% 
  filter(mean_relabund > 0.001) %>%
  select(days, z_interv, pco, mean_relabund, relacount) %>% 
  distinct() %>% 
  mutate(font_col = ifelse(mean_relabund < 0.03, "white", "black")) %>% 
  ggplot(aes(x = days, y = reorder(pco, mean_relabund))) +
   geom_tile(aes(fill = mean_relabund), color = "white") +
  # scale_color_manual(values = custom.colors) %>%
  # scale_fill_gradientn(colors = rev(matlab.colors)) +
  scale_fill_viridis_b(option = "D",  trans = 'log10') +
  geom_text(aes(label = round(mean_relabund, 2), color = font_col), size = 4) +
  scale_color_manual(values = c("white" = "white", "black" = "black")) +
  labs(x = "Days", y = "Phylum: Class: Order", fill = "Relative Abundance") +
  facet_grid(~factor(z_interv, levels = levels)) +
  theme_classic2(base_size = 16) +
  theme(axis.text.y = element_text(size = 12), legend.position = "top") +
   guides(fill = guide_colourbar(barheight = 2, barwidth = 20, frame.colour = "black", frame.linewidth = 2,ticks.colour = "black", ticks.linewidth = 1), color = F) +
  ggtitle("")


relacounts_combine_s4_order <- rela.counts_order %>% 
  filter(mean_relabund > 0.001) %>%
  select(days, z_interv, pco, mean_relabund, relacount, norm_count) %>% 
  distinct() %>% 
  mutate(font_col = ifelse(relacount < 1 * 10^10, "white", "black")) %>% 
  ggplot(aes(x = days, y = reorder(pco, relacount))) +
  geom_tile(aes(fill = norm_count), color = "white") +
  # scale_color_manual(values = custom.colors) %>%
  # scale_fill_gradientn(colors = rev(matlab.colors)) +
  scale_fill_viridis_b(option = "D",  trans = 'log10') +
 geom_text(aes(label = formatC(norm_count, format = "e", digits = 1), color = font_col), size = 4) +
  scale_color_manual(values = c("white" = "white", "black" = "black")) +
   labs(x = "Days", y = "Class : Order", fill = expression(paste("∆ Cells L"^-1))) +
  facet_grid(~factor(z_interv, levels = levels)) +
  theme_classic2(base_size = 16) +
  theme(axis.text.y = element_text(size = 12), legend.position = "top") +
   guides(fill = guide_colourbar(barheight = 2, barwidth = 20, frame.colour = "black", frame.linewidth = 2,ticks.colour = "black", ticks.linewidth = 1), color = F) +
  ggtitle("")



  

```

```{r}
del_data_combine_s4_order <- rela.counts_order  %>% 
  group_by(z_interv, pco) %>% 
   mutate(init_relabund = ifelse(days == 0, mean_relabund, NA),
         init_count = ifelse(days == 0, norm_count, NA)) %>% 
  fill(c(init_relabund, init_count), .direction = "updown") %>% 
  mutate(del_relabund = ifelse(days > 0, mean_relabund - init_relabund, 0),
         del_count = ifelse(days > 0, norm_count - init_count, 0)) %>% 
  ungroup()


```

```{r fig.height=8, fig.width=12}
relaheat_del_combine_s4_order <- del_data_combine_s4_order  %>%
  select(days, z_interv, pco, mean_relabund, del_relabund) %>% 
  distinct() %>% 
  filter(mean_relabund > 0.001, days > 0) %>%
  ggplot(aes(x = days, y = reorder(pco, mean_relabund))) +
  geom_tile(aes(fill = round(del_relabund, 2)), color = "white") +
  scale_fill_viridis_c() +
  geom_text(aes(label = round(del_relabund, 2), color = "white"), size = 4) +
  scale_color_manual(values = c("white" = "white", "black" = "black")) +
  labs(x = "Days", y = "", fill = "∆ Relative Abundance") +
  facet_grid(~factor(z_interv, levels = levels)) +
  theme_classic2(base_size = 16) +
  theme( legend.position = "top", axis.text.y = element_blank()) +
   guides(fill = guide_colourbar(barheight = 2, barwidth = 20, frame.colour = "black", frame.linewidth = 2,ticks.colour = "black", ticks.linewidth = 1), color = F) 


relacount_del_combine_s4_order <- del_data_combine_s4_order  %>%
  select(days, z_interv, pco, mean_relabund, del_relabund, del_count) %>% 
  distinct() %>% 
  filter(mean_relabund > 0.001, days > 0) %>%
  ggplot(aes(x = days, y = reorder(pco, mean_relabund))) +
  geom_tile(aes(fill = del_count), color = "white") +
  # scale_fill_gradient2(low = "#313695", mid = "#FEE090", high = "#A50026", midpoint = 0) +
  # scale_fill_gradientn(colors = rev(odv.colors)) +
   # scale_fill_viridis_b(begin = 0.2) +
  scale_fill_viridis_c(begin = 0.2) +
  geom_text(aes(label = formatC(del_count, format = "e", digits = 1), color = "white"), size = 4) +
  scale_color_manual(values = c("white" = "white", "black" = "black")) +
  labs(x = "Days", y = "", fill = expression(paste("∆ Cells L"^-1))) +
  facet_grid(~factor(z_interv, levels = levels)) +
  theme_classic2(base_size = 16) +
  theme( legend.position = "top", axis.text.y = element_blank()) +
  # theme( legend.position = "top") +
   guides(fill = guide_colourbar(barheight = 2, barwidth = 20, frame.colour = "black", frame.linewidth = 2,ticks.colour = "black", ticks.linewidth = 1), color = F) 




```

#### s6

```{r warning = F}

combine_s6 <- n3s6 %>% 
  transform_sample_counts(., function(x) x*1) #not actually transforming counts here...just converting frpm phyloseq object
 combine_s6.df <-  as(otu_table(combine_s6), "matrix") %>%
   as.data.frame(.) 


#then combine the data frames
custom.tab_combine_s6 <- tax.df %>% 
  rownames_to_column(., var = "asv") %>% 
  left_join(., combine_s6.df %>% rownames_to_column(., var = "asv")) %>% 
  #create a new index of that combines the  class, order, family, and genus values, you can play around here!!
  mutate(#pcofg = paste(Phylum, "_", Class, "_", Order,"_", Family, "_", Genus),
         pcof = paste(Phylum, "_", Class, "_", Order,"_", Family)) %>% 
         # pco = paste(Phylum, "_", Class, "_", Order)) %>% 
  select(-c(asv:Genus)) %>% 
  select(pcof,everything()) %>%
  group_by(pcof) %>%
  # select(pco,everything()) %>% 
  # group_by(pco) %>% 
  #here we are combining the relative abundances based on our grouping
  summarise_at(vars(!contains(c("pcof"))), sum, na.rm = T) %>% 
  ungroup()

#save the row names and then make them into the column names
colnames <- custom.tab_combine_s6[,1] 

#transpose the dataframe so we can merge with the sample info table
t_custom.tab_combine_s6 <-  as.data.frame(t(custom.tab_combine_s6[,-1]))
colnames(t_custom.tab_combine_s6) <- colnames$pcof
# colnames(t_custom.tab) <- colnames$pco

#merge
sweet.tab_custom.tab_combine_s6  <- t_custom.tab_combine_s6  %>% 
  rownames_to_column(., var = "sample") %>% 
  left_join(., sample.tab %>% rownames_to_column(., var = "sample")) %>% 
  select(sample, Cruise:npp, everything())


counts_custom.tab_combine_s6  <- sweet.tab_custom.tab_combine_s6 %>% 
  select(-c(sample:npp)) %>% 
  #remove groups that are completely absent
  .[ , colSums(.) > 0] %>% 
  #arrange by biggest contributors
  .[, order(colSums(-.))] %>% 
  bind_cols(sweet.tab_custom.tab_combine_s6 %>% select(sample:npp), .) %>% 
  mutate_at(vars(days), as.character) %>% 
  group_by(days,  z_interv) %>% 
  summarise_at(vars(!c(sample:npp)), sum, na.rm = T) %>% 
  ungroup() %>% 
  rowwise() %>% 
  mutate(total = sum(c_across(where(is.numeric)))) %>% 
  ungroup() %>% 
  select(days:z_interv, total, everything())  %>% 
  mutate(across(c(4:169), ~ . / total)) %>% 
  select(-total)
  
```

```{r echo = FALSE, warning = FALSE, message = FALSE, fig.height = 8, fig.width = 12, fig.align = "center"}
relaheat.data_combine_s6 <- counts_custom.tab_combine_s6 %>% 
  pivot_longer(.,-c(days:z_interv), names_to = "taxa", values_to = "relabund") %>% 
  separate(taxa, into = c("p", "c", "o", "f"), sep = " _ ") %>% 
  mutate(pcof = paste(p, c, o, f, sep = ": ")) %>% 
  group_by(days, z_interv, pcof) %>% 
  mutate(mean_relabund = mean(relabund, na.rm = T)) %>% 
  ungroup() %>% 
  select(-relabund) %>% 
  distinct() %>% 
  mutate_at(vars(p,c,o, pcof), str_replace_all, "NA", "Unassigned") %>% 
  arrange(desc(pcof), mean_relabund)


relaheat_combine_s6 <- relaheat.data_combine_s6 %>%
 filter(mean_relabund > 0.001) %>%
  select(days, z_interv, pcof, mean_relabund) %>% 
  distinct() %>% 
  ggplot(aes(x = days, y = reorder(pcof, mean_relabund))) +
  geom_tile(aes(fill = mean_relabund), color = "white") +
  # scale_color_manual(values = custom.colors) %>%
  # scale_fill_gradientn(colors = rev(matlab.colors)) +
  scale_fill_viridis_b(option = "D",  trans = 'log10') +
  geom_text(aes(label = round(mean_relabund, 3), color = "white"), size = 2) +
  scale_color_manual(values = c("white" = "white", "black" = "black")) +
  labs(x = "Days", y = "Class: Order: Family", fill = "Relative Abundance") +
  facet_grid(~factor(z_interv, levels = levels)) +
  theme_classic2(base_size = 16) +
  theme(axis.text.y = element_text(size = 12), legend.position = "top") +
   guides(fill = guide_colourbar(barheight = 2, barwidth = 20, frame.colour = "black", frame.linewidth = 2,ticks.colour = "black", ticks.linewidth = 1), color = F) +
  ggtitle("NAAMES 3 Station 6")


```

```{r}
del_data_combine_s6 <- relaheat.data_combine_s6  %>% 
  group_by(z_interv, pcof) %>% 
  mutate(init_relabund = ifelse(days == 0, mean_relabund, NA)) %>% 
  fill(init_relabund, .direction = "updown") %>% 
  mutate(del_relabund = ifelse(days > 0, mean_relabund - init_relabund, 0)) %>% 
  ungroup()
```


```{r fig.height=8, fig.width=12}
relaheat_del_combine_s6 <- del_data_combine_s6  %>%
  filter(mean_relabund > 0.001, days > 0) %>%
  select(days, z_interv, pcof, mean_relabund, del_relabund) %>% 
  distinct() %>% 
  ggplot(aes(x = days, y = reorder(pcof, mean_relabund))) +
  geom_tile(aes(fill = del_relabund), color = "white") +
  # scale_fill_gradient2(low = "#313695", mid = "#FEE090", high = "#A50026", midpoint = 0) +
  # scale_fill_gradientn(colors = rev(odv.colors)) +
  scale_fill_viridis_b(option = "D") +
  scale_fill_viridis_c() +
   geom_text(aes(label = round(del_relabund, 3), color = "white"), size = 2) +
  scale_color_manual(values = c("white" = "white", "black" = "black")) +
  labs(x = "Days", y = "", fill = "∆ Relative Abundance") +
  facet_grid(~factor(z_interv, levels = levels)) +
  theme_classic2(base_size = 16) +
  theme( legend.position = "top", axis.text.y = element_blank()) +
  # theme( legend.position = "top") +
   guides(fill = guide_colourbar(barheight = 2, barwidth = 20, frame.colour = "black", frame.linewidth = 2,ticks.colour = "black", ticks.linewidth = 1), color = F) 

```

#### combine

```{r fig.height=12, fig.width=21}
(relaheat_combine_s4_order + relaheat_del_combine_s4_order) 
```

```{r fig.height=12, fig.width=21}
(relacounts_combine_s4_order + relacount_del_combine_s4_order) 
```


```{r fig.height=12, fig.width=21}
(relaheat_combine_s4 + relaheat_del_combine_s4) 
```

```{r fig.height=12, fig.width=21}
(relacounts_combine_s4 + relacount_del_combine_s4) 
```


```{r fig.height=14, fig.width=21}
(relaheat_combine_s6 + relaheat_del_combine_s6) 
```

```{r fig.height=12, fig.width=21}
relaheat_combine_s4 + (relacounts_combine_s4 + labs(y = "") + theme( legend.position = "top", axis.text.y = element_blank()) )
```

## taxa line plots

```{r fig.height=6, fig.width=12}
levels2 <- c("SAR11_Ia", "Flavobacteriaceae", "ZD0405", "SAR86_clade", "SAR11_II", "SAR202_Clade1", "OM1_clade", "Salinisphaeraceae", "OCS116_clade")

taxa_sp.data <- rela.counts %>% 
  select(-sd_relabund) %>% 
  filter(o %in% c("OCS116_clade") | f %in% c("SAR202_Clade1","Flavobacteriaceae",  "SAR11_Ia", "SAR11_II", "OM1_clade", "Salinisphaeraceae", "ZD0405", "SAR86_clade")) %>% 
  group_by(z_interv, pcof) %>% 
  mutate(first_relabund = ifelse(days == 0, mean_relabund, NA),
         first_count = ifelse(days == 0, norm_count, NA)) %>% 
  fill(first_relabund, first_count) %>% 
  mutate(del_relabund = mean_relabund - first_relabund,
         del_count = norm_count - first_count) %>% 
  ungroup() %>% 
  mutate_at(vars(contains("relabund")), funs(.*100)) %>% 
  mutate(f = ifelse(f == "NA", "OCS116_clade", f))

most.taxa <- taxa_sp.data %>% 
  filter(!f %in% c( "SAR11_II", "SAR202_Clade1", "OM1_clade", "Salinisphaeraceae", "OCS116_clade")) %>% 
  filter(z_interv != "Euphotic") %>% 
  ggplot(aes(x = days, y = norm_count, group = interaction(z_interv, f))) +
  geom_line(color = "#377EB8", size = 0.7) +
  geom_point(shape = 21, size = 6, fill = "#377EB8", color = "black", stroke = 1) + 
  scale_y_continuous(trans = 'log10') +
  labs(x = "Days", y = expression(paste("Log Cell Abundance, L"^-1)), colour = "", fill = "Days") +

  theme_classic2(base_size = 20) +
  facet_grid(~ factor(f, levels = levels2)) 
  
  
```

```{r fig.height=6, fig.width=12}
most.taxa2 <- taxa_sp.data %>% 
  filter(!f %in% c( "SAR11_II", "SAR202_Clade1", "OM1_clade", "Salinisphaeraceae", "OCS116_clade")) %>% 
  filter(z_interv != "Euphotic") %>% 
  ggplot(aes(x = days, y = del_relabund, group = interaction(z_interv, f))) +
  geom_line(size = 0.7) +
  geom_point(shape = 21, size = 6, fill = "white", color = "black", stroke = 1) + 

  labs(x = "Days", y = expression(paste("∆ Relative Abundance, %")), colour = "", fill = "Days") +
  theme_classic2(base_size = 20) +
  facet_grid(~ factor(f, levels = levels2)) 
```


```{r fig.width=12, fig.height= 5}
mz.taxa2 <- taxa_sp.data %>% 
  filter(!z_interv == "Euphotic", !f %in% c("SAR11_Ia", "Flavobacteriaceae", "ZD0405", "SAR86_clade")) %>% 
  ggplot(aes(x = days, y = norm_count, group = interaction(z_interv, pcof))) +
  geom_line(color = "#377EB8", size = 0.7) +
  geom_point(shape = 21, size = 6, fill = "#377EB8", color = "black", stroke = 1) + 
  labs(x = "Days", y = expression(paste("Log Cell Abundance, L"^-1)), colour = "", fill = "Days") +
  scale_y_continuous(trans = 'log10') +
  theme_classic2(base_size = 20) +
  facet_grid( ~ factor(f, levels = levels2))
  
```




```{r fig.width=14, fig.height=6, }
mz.taxa <- taxa_sp.data %>% 
  filter(!z_interv == "Euphotic", !f %in% c("SAR11_Ia", "Flavobacteriaceae", "ZD0405", "SAR86_clade")) %>% 
  ggplot(aes(x = days, y = mean_relabund, group = interaction(z_interv, pcof))) +
  geom_line(size = 0.7) +
  geom_line(data = taxa_sp.data %>% filter(!z_interv == "Euphotic", !f %in% c("SAR11_Ia", "Flavobacteriaceae", "ZD0405", "SAR86_clade")), aes(x = days, y = norm_count/10^7, group = interaction(z_interv, pcof)), color = "#377EB8") +
  geom_point(data = taxa_sp.data %>% filter(!z_interv == "Euphotic", !f %in% c("SAR11_Ia", "Flavobacteriaceae", "ZD0405", "SAR86_clade")), aes(x = days, y = norm_count/10^7, group = interaction(z_interv, pcof)), shape = 21, size = 6, fill = "#377EB8", color = "black", stroke = 1) + 
  geom_point(shape = 21, size = 6, fill = "white", color = "black", stroke = 1) + 
  labs(x = "Days", y = expression(paste("Relative Abundance, %")), colour = "", fill = "Days") +
  
  scale_y_continuous(
    # Add a second axis and specify its features
    sec.axis = sec_axis(~.*10^7, name = expression(paste("Cell Abundance, L"^-1)))
  ) + 
  
  # scale_fill_viridis() +
  # scale_color_viridis() +
  # guides(fill = F, color = F, linetype = F, shape = F) +
  theme_classic2(base_size = 20) +
  facet_grid( ~ factor(f, levels = levels2)) +
  theme(axis.title.y.right = element_text(color = "#377EB8")) ; mz_taxa
  
```



# Plots for powerpoint

```{r}
library(officer)
```


```{r}


p1a <- alpha.plot_n2s4 + 
   guides(colour = guide_colorbar(reverse = T),
         fill = guide_colorbar(reverse = T, barheight = 8, barwidth = 3, frame.colour = "black", frame.linewidth = 2,ticks.colour = "black", ticks.linewidth = 1)) + theme_classic2(16) + theme(axis.title.x = element_blank(), axis.text.x = element_blank()) 
p1b <- alpha.plot_n3s6 + theme_classic2(16)
# p1c <- alpha.plot_s4_core_s6_mz + theme_classic2(16)
  
p1 <-  p1a / p1b  + plot_layout(guides = 'collect')  + plot_annotation(tag_levels = "a") &
  theme(plot.tag = element_text(size = 22)) 

###

p2a <- nmds_s4s6_core.plot + 
  guides( shape = guide_legend(title = "Depth Horizon"), color = F,  linetype = guide_legend(title = "Depth Horizon"), fill = guide_legend(title = "Cruise", override.aes = list(shape = 21))) + theme_classic2(16)
p2b <- nmds_n2s4_core.plot + 
  guides(fill = guide_colorbar(title = "Days", barheight = 8, barwidth = 2, frame.colour = "black", frame.linewidth = 2,ticks.colour = "black", ticks.linewidth = 1, reverse = T), shape = guide_legend(title = "Depth Horizon"), linetype = guide_legend(title = "Days"), color = F) + theme_classic2(16)
p2c <- nmds_n3s6.plot + 
  guides(fill = guide_colorbar(title = "Days", barheight = 8, barwidth = 2, frame.colour = "black", frame.linewidth = 2,ticks.colour = "black", ticks.linewidth = 1, reverse = T), shape = guide_legend(title = "Depth Horizon"),  linetype = guide_legend(title = "Depth Horizon"), color = F) + theme_classic2(16)
  
p2 <- p2a + p2b + p2c   + plot_annotation(tag_levels = "a") &
  theme(plot.tag = element_text(size = 22)) 

###



p3a <- relaheat_combine_s4 + theme_classic2(16) + theme( legend.position = "top") +
  # theme( legend.position = "top") +
   guides(fill = guide_colourbar(barheight = 2, barwidth = 20, frame.colour = "black", frame.linewidth = 2,ticks.colour = "black", ticks.linewidth = 1), color = F) 
p3b <- relacounts_combine_s4 +  labs(y = "") +  theme_classic2(16) + theme( legend.position = "top", axis.text.y = element_blank()) +
  # theme( legend.position = "top") +
   guides(fill = guide_colourbar(barheight = 2, barwidth = 20, frame.colour = "black", frame.linewidth = 2,ticks.colour = "black", ticks.linewidth = 1), color = F) 

p3 <- p3a + p3b  + plot_annotation(tag_levels = "a") &
  theme(plot.tag = element_text(size = 22)) 


###


p4a <- relaheat_combine_s6 + theme_classic2(16) + theme( legend.position = "top") +
  # theme( legend.position = "top") +
   guides(fill = guide_colourbar(barheight = 2, barwidth = 20, frame.colour = "black", frame.linewidth = 2,ticks.colour = "black", ticks.linewidth = 1), color = F) 
p4b <- relaheat_del_combine_s6 + theme_classic2(16) + theme( legend.position = "top", axis.text.y = element_blank()) +
  # theme( legend.position = "top") +
   guides(fill = guide_colourbar(barheight = 2, barwidth = 20, frame.colour = "black", frame.linewidth = 2,ticks.colour = "black", ticks.linewidth = 1), color = F) 

p4 <- p4a + p4b  + plot_annotation(tag_levels = "a") &
  theme(plot.tag = element_text(size = 22)) 




####

p6a <-  nmds_n2s4_core.plot + 
  guides(fill = guide_colorbar(title = "Days", barheight = 8, barwidth = 2, frame.colour = "black", frame.linewidth = 2,ticks.colour = "black", ticks.linewidth = 1, reverse = T), shape = guide_legend(title = "Depth Horizon"), linetype = guide_legend(title = "Days"), color = F) + theme_classic2(16)
p6b <- nmds_n3s6.plot + 
  guides(fill = guide_colorbar(title = "Days", barheight = 8, barwidth = 2, frame.colour = "black", frame.linewidth = 2,ticks.colour = "black", ticks.linewidth = 1, reverse = T), shape = guide_legend(title = "Depth Horizon"),  linetype = guide_legend(title = "Depth Horizon"), color = F) + theme_classic2(16)
  
p6 <- p6b + p6a + plot_annotation(tag_levels = "a") &
  theme(plot.tag = element_text(size = 22)) 


  
# initialize PowerPoint slide
officer::read_pptx() %>%
  # add slide ----
  officer::add_slide() %>% 
  # specify object and location of object 
  officer::ph_with(p1, ph_location(width = 11, height = 9)) %>%
  
  officer::add_slide() %>% 
  # specify object and location of object 
  officer::ph_with(p2, ph_location(width = 20, height = 6)) %>%
  
  officer::add_slide() %>% 
  # specify object and location of object 
  officer::ph_with(p3, ph_location(width = 21, height = 12)) %>%
  
  officer::add_slide() %>% 
  # specify object and location of object 
  officer::ph_with(p4, ph_location(width = 21, height = 12)) %>%
  
  
  officer::add_slide() %>% 
  # specify object and location of object 
  officer::ph_with(p6, ph_location(width = 16, height = 8)) %>%
  
   officer::add_slide() %>% 
  # specify object and location of object 
  officer::ph_with(mz.taxa2, ph_location(width = 12, height = 5)) %>%
  
   officer::add_slide() %>% 
  # specify object and location of object 
  officer::ph_with(most.taxa, ph_location(width = 12, height = 5)) %>%
  
  
   officer::add_slide() %>% 
  # specify object and location of object 
  officer::ph_with(most.taxa2, ph_location(width = 12, height = 5)) %>%
  
     officer::add_slide() %>% 
  # specify object and location of object 
  officer::ph_with(mz.taxa, ph_location(width = 14, height = 6)) %>%
  
  
  # export slide 
  base::print(
    target = "~/Desktop/Dissertation/MS_N2S4/Presentations/16s_aug18.pptx"
    )

```





# Save

```{r}

saveRDS(sample.tab, "~/GITHUB/naames_multiday/Output/Sample_Table.rds")
saveRDS(relabund, "~/GITHUB/naames_multiday/Output/Relabund_Table.rds")
saveRDS(relabund_s6, "~/GITHUB/naames_multiday/Output/Relabund_Table_s6.rds")
saveRDS(sub_ps, "~/GITHUB/naames_multiday/Output/phyloseq_obj.rds")
saveRDS(ps_min, "~/GITHUB/naames_multiday/Output/phyloseq_obj_min.rds")


```
















