---
title: "16S_revised"
author: "Nicholas Baetge"
date: "12/28/2021"
output: github_document
---

# Intro


```{r load libraries, message=FALSE, warning=FALSE}
library(tidyverse)
library(lubridate)
library(hms)
library(zoo) 
library(oce)  
library(ggpubr)
library(patchwork)
library(phyloseq)
```

# Import Data 

```{r}
floats <- read_rds("~/GITHUB/naames_multiday/Output/processed_floats.rds") 

floats %>% select(float:lon, ezd) %>% distinct() %>% drop_na(ezd) %>% 
  group_by(Cruise, Station) %>% 
  summarize(max_ezd = max(ezd))
  
```
Euphotic zone for N2S4 = 75 m, for N3 is < 75 m (don't know for station 3.5, but likely similar to station 3 and 4)

```{r load data, message=FALSE, warning=FALSE}

count.tab <- read.table("~/GITHUB/naames_multiday/Input/16s/HetV1OTU.txt", header = T, row.names = 1, check.names = F) 

tax.tab <- as.matrix(read.table("~/GITHUB/naames_multiday/Input/16s/HetV1TUtax.txt", header = T, row.names = 1, check.names = F, na.strings = "", sep = "\t"))

sample.tab <- read_rds("~/GITHUB/naames_multiday/Input/bottle_data.rds") %>% 
  drop_na(DNA_ID) %>% 
  filter(Cruise == "AT34" & Station == 4 | Cruise == "AT38" & Station %in% c(3, 3.5, 4, 6))   %>% 
  filter(!plot_date %in% c("May 24 02:30", "May 27 06:07", "Sep 14 15:33", "Sep 15 03:04", "Sep 15 15:20", "Sep 16 03:04", "Sep 16 04:50", "Sep 16 07:26", "Sep 8 03:08", "Sep 8 03:08",  "Sep 8 15:30", "Sep 11 03:07"), z <= 200) %>% 
  mutate(dh = ifelse(Cruise == "AT34" & z < 100, "Euphotic", "Upper Mesopelagic"),
         dh = ifelse(Cruise == "AT38" & z < 75, "Euphotic", dh),
         Cruise = ifelse(Cruise == "AT34", "NAAMES 2", "NAAMES 3"),
         cruise_station = paste(Cruise, "Station", Station)) %>% 
  column_to_rownames(var = "DNA_ID") %>% 
  select(Cruise:mld, dh, everything())


gene_copies <- readxl::read_xlsx("~/GITHUB/naames_multiday/Input/16s/gene_copies.xlsx") %>% 
  mutate(copy_num = ave_order, 
         copy_num = ifelse(is.na(copy_num), ave_class, copy_num),
         copy_num = ifelse(is.na(copy_num), ave_phylum, copy_num),
         copy_num = ifelse(is.na(copy_num), ave_kingdom, copy_num))

```


# Phyloseq Object

We need to create a phyloseq object that merges all three datasets. Sometimes this doesn't work beacuse of the format of the data files. Make sure all the sample names between the sampleinfo.txt and seqtab-nochimtaxa.txt are the same

```{r}
OTU = otu_table(count.tab, taxa_are_rows = TRUE) 
TAX = tax_table(tax.tab)
SAM = sample_data(sample.tab)
ps = phyloseq(OTU,TAX,SAM) 

```

# Filter sequences

We will filter out  chloroplasts and mitochondria, because we only intended to amplify bacterial sequences. It's good to check you don’t have anything lurking in the taxonomy table. 

```{r}
sub_ps <- ps %>%
  subset_taxa(
    Family  != "mitochondria" &
    Order   != "Chloroplast")
```

# Sample Summary

As a first analysis, we will look at the distribution of read counts from our samples

```{r echo = FALSE, warning = FALSE, message = FALSE, fig.height = 6, fig.width = 12, fig.align = "center"}
# Make a data frame with a column for the read counts of each sample
sample_sum_df <- data.frame(sum = sample_sums(sub_ps))

# Histogram of sample read counts
ggplot(sample_sum_df, aes(x = sum)) + 
  geom_histogram(color = "black", fill = "#377EB8", binwidth = 2500) +
  ggtitle("Distribution of sample sequencing depth") + 
  xlab("Read counts") +
  theme(axis.title.y = element_blank()) +
  theme_linedraw(base_size = 16)
```

```{r}
# mean, max and min of sample read counts
smin <- min(sample_sums(sub_ps)) 
smean <- mean(sample_sums(sub_ps)) 
smax <- max(sample_sums(sub_ps)) 
```

# Beta Diversity

Beta diversity involves calculating metrics such as distances or dissimilarities based on pairwise comparisons of samples – they don’t exist for a single sample, but rather only as metrics that relate samples to each other. i.e. beta diversity =  patterns in community structure between samples

Since differences in sampling depths between samples can influence distance/dissimilarity metrics, we first need to somehow normalize the read depth across our samples. 

## Subsample

We will rarefy (random subsample with replacement) the min read depth of the samples first (scale to the smallest library size) according to Bolaños et al 2021.

A strong reason to subsample is to standardize effort. The bottom line is that in all experimental design you should not be comparing things to which you devote different effort in resolution. For instance, you don't sample one site once a week and another once a month if you want to compare the dynamics between the sites. You standardize effort.


```{r}

ps_min <-  rarefy_even_depth(sub_ps, sample.size = smin, rngseed = 532898)

```


We can also subset the N2 and the N2S4 data here


```{r}
s4 <- ps_min %>% 
  subset_samples(Cruise == "NAAMES 2" & Station == 4)
```


```{r}
s6 <- ps_min %>% 
  subset_samples(Cruise == "NAAMES 3" & Station == 6)
```

```{r}
n3 <- ps_min %>% 
  subset_samples(Cruise == "NAAMES 3")
```



## Unconstrained Ordination

One of the best exploratory analyses for amplicon data is unconstrained ordinations. Here we will look at ordinations of our subsampled dataset

### NMDS

Let’s try an NMDS. For NMDS plots it’s important to set a seed since the starting positions of samples in the alogrithm is random.

```{r}
set.seed(8)
# Ordinate
nmds <- ordinate(ps_min, method = "NMDS",  distance = "bray") # stress = 0.07
```


```{r}
set.seed(7)
# Ordinate
nmds_s4 <- ordinate(s4, method = "NMDS",  distance = "bray") # stress = 0.09
```

```{r}
set.seed(7)
# Ordinate
nmds_n3 <- ordinate(n3, method = "NMDS",  distance = "bray") # stress = 0.05
```


NMDS plots attempt to show ordinal distances between samples as accurately as possible in two dimensions. It is important to report the stress of these plots, because a high stress value means that the algorithm had a hard time representing the distances between samples in 2 dimensions. The stress of these plots were good - it was .1 (generally anything below .2 is considered acceptable).


```{r echo = FALSE, warning = FALSE, message = FALSE, fig.height = 8, fig.width = 10, fig.align = "center"}

nmds.plot <- plot_ordination(ps_min, nmds,  title = "NAAMES 2 Station 4 & NAAMES 3") +
  geom_point(aes(fill = cruise_station, shape = dh), alpha = 0.7, size = 4) +
  stat_ellipse(aes(group = cruise_station), color = "firebrick", level = 0.95, type = "t", size = 0.7) +
  scale_shape_manual(values = c(21, 22)) +
  scale_fill_viridis_d(end = 0.7) +
  theme_linedraw(base_size = 16) +
  guides( shape = guide_legend(title = "Depth Horizon"), color = F,  linetype = guide_legend(title = "Depth Horizon"), fill = guide_legend(title = "", override.aes = list(shape = 21))) 
#removing one of the plotting layers (there are points within points)
nmds.plot$layers <- nmds.plot$layers[-1]
```


```{r echo = FALSE, warning = FALSE, message = FALSE, fig.height = 8, fig.width = 10, fig.align = "center"}

nmds_s4.plot <- plot_ordination(s4, nmds_s4,  title = "NAAMES 2 Station 4") +
  geom_point(aes(fill = plot_date, shape = dh), alpha = 0.7, size = 4) +
  stat_ellipse(aes(group = plot_date), level = 0.95, color = "firebrick", type = "t") +
  scale_shape_manual(values = c(21, 22)) +
  # scale_color_manual(values = custom.colors) +
  scale_fill_viridis_d(end = 0.7) +
  theme_linedraw(base_size = 16) +
  guides( shape = guide_legend(title = "Depth Horizon"), color = F,  linetype = guide_legend(title = ""), fill = guide_legend(title = "", override.aes = list(shape = 21))) 
#removing one of the plotting layers (there are points within points)
nmds_s4.plot$layers <- nmds_s4.plot$layers[-1]

```


```{r echo = FALSE, warning = FALSE, message = FALSE, fig.height = 8, fig.width = 10, fig.align = "center"}

nmds_n3.plot <- plot_ordination(n3, nmds_n3,  title = "NAAMES 3") +
  geom_point(aes(fill = cruise_station, shape = dh), alpha = 0.7, size = 4) +
  stat_ellipse(aes(group = dh), level = 0.95, color = "firebrick", type = "t") +
  scale_shape_manual(values = c(21, 22)) +
  # scale_color_manual(values = custom.colors) +
  scale_fill_viridis_d(end = 0.7) +
  theme_linedraw(base_size = 16) +
  guides( shape = guide_legend(title = "Depth Horizon"), color = F,  linetype = guide_legend(title = ""), fill = guide_legend(title = "", override.aes = list(shape = 21), order = 2)) 
#removing one of the plotting layers (there are points within points)
nmds_n3.plot$layers <- nmds_n3.plot$layers[-1]

```

```{r fig.height=6, fig.width=23}
nmds.plot + nmds_s4.plot + nmds_n3.plot +  
  plot_annotation(tag_levels = "a") &
  theme(plot.tag = element_text(size = 22),
        plot.title = element_text(size = 18)) 
```


95% confidence interval ellipse for the mean (group centroid) tells us something about the sampling distribution of the mean (centroid) we might see if we repeated your data collection a lot of times. In other words we are looking at the uncertainty in the estimate of the population mean (centroid) given the sample of data we collected.


# Betadisper and permutational ANOVA

Above, we performed beta diversity analyses on Bray-Curtis distances on
rarefied datasets that were then visualized using NMDS.
We can test if there are statistically significant differences between
sample groups using the betadisper and adonis functions of the vegan
package. Betadisper tests whether two or more groups are homogeneously
dispersed in relation to their species in studied samples. This test can be done to see if one group has more compositional variance than
another. Moreover, homogeneity of dispersion among groups is very
advisable to have if you want to test if two or more groups have
different compositions, which is tested by adonis.

## Phyloseq to DESeq, distance matrix

To be able to run the stats, we first have to create a distance matrix
from our data. We’ll use the DESeq package to do so.

```{r message=FALSE, warning=FALSE}
library(DESeq2)
library(vegan)
```

```{r}
deseq_counts <- phyloseq_to_deseq2(ps_min, design = ~datetime ) #the design argument is required but doesn't matter here
```

```{r}
deseq_count_tab <- assay(deseq_counts) #extract the read count matrix
```

We’ll calculate bray-curtis distances, which reflect the NMDS ordinations above


```{r}
#We can subset our data if we want to and calculate distances/run stats for only a subset of the group. The code below shows how

sample.tab2 <- sample.tab %>%
  rownames_to_column() %>% 
  column_to_rownames(var = "rowname")


####

subset_sample_IDs_n2s4 <-  row.names(sample.tab2)[sample.tab2$Cruise == "NAAMES 2" & sample.tab2$Station == 4]

dist_n2s4 <- vegdist(t(deseq_count_tab[ , colnames(deseq_count_tab) %in% subset_sample_IDs_n2s4]), method = "bray")

sample_info_tab_n2s4 <- sample.tab2[row.names(sample.tab2) %in% subset_sample_IDs_n2s4, ]

####

subset_sample_IDs_n2s4_ez <-  row.names(sample.tab2)[sample.tab2$Cruise == "NAAMES 2" & sample.tab2$Station == 4 & sample.tab2$dh == "Euphotic"]

dist_n2s4_ez <- vegdist(t(deseq_count_tab[ , colnames(deseq_count_tab) %in% subset_sample_IDs_n2s4_ez]), method = "bray")

sample_info_tab_n2s4_ez <- sample.tab2[row.names(sample.tab2) %in% subset_sample_IDs_n2s4_ez, ]

####


subset_sample_IDs_n2s4_mz <-  row.names(sample.tab2)[sample.tab2$Cruise == "NAAMES 2" & sample.tab2$Station == 4 & sample.tab2$dh == "Upper Mesopelagic"]

dist_n2s4_mz <- vegdist(t(deseq_count_tab[ , colnames(deseq_count_tab) %in% subset_sample_IDs_n2s4_mz]), method = "bray")

sample_info_tab_n2s4_mz <- sample.tab2[row.names(sample.tab2) %in% subset_sample_IDs_n2s4_mz, ]


####

subset_sample_IDs_n3 <-  row.names(sample.tab2)[sample.tab2$Cruise == "NAAMES 3"]

dist_n3 <- vegdist(t(deseq_count_tab[ , colnames(deseq_count_tab) %in% subset_sample_IDs_n3]), method = "bray")

sample_info_tab_n3 <- sample.tab2[row.names(sample.tab2) %in% subset_sample_IDs_n3, ]

```


Betadisper first calculates the average distance of group members to the group centroid in multivariate space (generated by a distance matrix).

In the function below: we are using the distance matrix to calculate the multivariate dispersions (variances; average distance to centroids). We then use group dispersions to perform an ANOVA test.

```{r}
anova(betadisper(dist_n2s4, sample_info_tab_n2s4$dh)) 
```


```{r}
anova(betadisper(dist_n2s4, sample_info_tab_n2s4$date)) 
```

```{r}
anova(betadisper(dist_n2s4_ez, sample_info_tab_n2s4_ez$date)) 
```

```{r}
anova(betadisper(dist_n3, sample_info_tab_n3$Station)) 
```

```{r}
anova(betadisper(dist_n3, sample_info_tab_n3$dh)) 
```

```{r}
anova(betadisper(dist_n2s4_mz, sample_info_tab_n2s4_mz$datetime)) 
```

The ANOVA’s p-value is not significant meaning that group dispersions are homogenous (“Null hypothesis of no difference in dispersion between groups”)


**Homogeneity among group dispersions (compositions vary similarly) for N2S4 with grouping as depth horizon, for N2S4 with grouping as date, for N2S4 euphotic zone with grouping as date, for N3 with grouping as Cruise and Station, for N3 with grouping as depth horizon. Compositions fo not vary similarly for N2S4 mesopelagic zone with grouping as date.** 


Adonis analyzes and partitions sums of squares using distance matrices. It can be seen as an ANOVA using distance matrices (analogous to MANOVA – multivariate analysis of variance). Therefore, it is used to test if two or more groups have similar compositions.

```{r}
adonis(dist_n2s4_mz~sample_info_tab_n2s4_mz$datetime)
```

```{r}
adonis(dist_n3~sample_info_tab_n3$dh)
```

**N2S4 mesopelagic zone, group = date: no homogeneity among group dispersions (compositions vary similarly) and are significantly different  **

# Alpha Diversity

We are going to calculate the the Chao1 diversity index. 

**it is important to note that the alpha diversity values are not interpretable as “real” numbers of anything (due to the nature of amplicon data), but they can still be useful as relative metrics of comparison.**

[Chao1: nonparametric estimation of minimum community richness](https://www.jstor.org/stable/4615964?seq=1#metadata_info_tab_contents) 

```{r message=FALSE, warning=FALSE}
richness <- estimate_richness(ps_min, measures = c("Chao1", "Shannon")) %>% 
  rownames_to_column(., var = "DNA_ID") %>% 
   separate(DNA_ID, into = c("a", "b"), sep = "_") %>% 
  separate(a, into = c("x", "y", "z")) %>% 
  mutate(z = ifelse(is.na(z), y, z),
         y = ifelse(y == z, y, NA), 
         x = ifelse(is.na(y), paste(x, "-5", sep = ""), x)) %>% 
  mutate(DNA_ID = paste(x, "-", z, "_", b, sep = "")) %>% 
  select(-c(x,y,z,b)) %>% 
  mutate(DNA_ID = ifelse(DNA_ID == "N3S3-5_S56", "N3S3-5-5_S56", DNA_ID))
```

```{r}
alphadiv <- left_join(richness, sample.tab %>% rownames_to_column(., var = "DNA_ID")) 
```


```{r echo = FALSE, warning = FALSE, message = FALSE, fig.height = 6, fig.width = 12, fig.align = "center"}

alphadiv.plot <- alphadiv %>% 
  mutate_at(vars(z), as.factor) %>% 
  mutate(Cruise = ifelse(Cruise == "NAAMES 2", "NAAMES 2 Station 4", Cruise)) %>% 
  ggplot(aes(x = factor(dh), y = Shannon)) +
  geom_boxplot(width = 0.3, outlier.shape = NA) +
  facet_grid(~Cruise, scales = "free") +
  geom_point(aes(fill = z), shape = 21, size = 3, alpha = 0.7) +
  scale_color_viridis_d(option = "viridis", direction = -1) +
  scale_fill_viridis_d(option = "viridis", direction = -1) +
  theme_linedraw(base_size = 16) +
  guides(fill = "none") +
  labs(x = "", y =  "Shannon", fill = "Depth, m") + 
  stat_compare_means(method = "t.test") 

chao.plot <- alphadiv %>% 
  mutate_at(vars(z), as.factor) %>% 
  mutate(Cruise = ifelse(Cruise == "NAAMES 2", "NAAMES 2 Station 4", Cruise)) %>% 
  ggplot(aes(x = factor(dh), y = Chao1)) +
  geom_boxplot(width = 0.3, outlier.shape = NA) +
  facet_grid(~Cruise, scales = "free") +
  geom_point(aes(fill = z), shape = 21, size = 3, alpha = 0.7) +
  scale_color_viridis_d(option = "viridis", direction = -1) +
  scale_fill_viridis_d(option = "viridis", direction = -1) +
  theme_linedraw(base_size = 16) +
  labs(x = "", y =  "Chao1", fill = "Depth, m") + 
  stat_compare_means(method = "t.test") 


```



```{r echo = FALSE, warning = FALSE, message = FALSE, fig.height = 6, fig.width = 12, fig.align = "center"}

my_comparisons <- list(c("May 24 06:10","May 25 06:05"),c("May 24 06:10","May 26 06:09"),c("May 25 06:05","May 26 06:09"))


n2s4_alphadiv.plot <- alphadiv %>% 
  filter(Cruise == "NAAMES 2", Station == 4) %>% 
  mutate_at(vars(z), as.factor) %>% 
  ggplot(aes(x = factor(plot_date), y = Shannon)) +
  geom_boxplot(width = 0.3, outlier.shape = NA) +
  facet_grid(~dh, scales = "free") +
  geom_point(aes(fill = z), shape = 21, size = 3, alpha = 0.7) +
  scale_color_viridis_d(option = "viridis", direction = -1) +
  scale_fill_viridis_d(option = "viridis", direction = -1) +
  theme_linedraw(base_size = 16) +
  guides(fill = "none") +
  labs(x = "", y =  "Shannon", fill = "Depth, m") + 
  stat_compare_means(comparisons = my_comparisons, method = "t.test") 

n2s4_chao.plot <- alphadiv %>% 
  filter(Cruise == "NAAMES 2", Station == 4) %>% 
  mutate_at(vars(z), as.factor) %>% 
  ggplot(aes(x = factor(plot_date), y = Chao1)) +
  geom_boxplot(width = 0.3, outlier.shape = NA) +
  facet_grid(~dh, scales = "free") +
  geom_point(aes(fill = z), shape = 21, size = 3, alpha = 0.7) +
  scale_color_viridis_d(option = "viridis", direction = -1) +
  scale_fill_viridis_d(option = "viridis", direction = -1) +
  theme_linedraw(base_size = 16) +
  labs(x = "", y =  "Chao1", fill = "Depth, m") + 
  stat_compare_means(comparisons = my_comparisons, method = "t.test") 


```

```{r fig.height=12, fig.width=14}
alphadiv.plot / chao.plot  +
  plot_layout(guides = "collect") +
  plot_annotation(tag_levels = "a") &
  theme(plot.tag = element_text(size = 22),
        plot.title = element_text(size = 18)) 

```

```{r fig.height=12, fig.width=14}
n2s4_alphadiv.plot / n2s4_chao.plot  +
  plot_layout(guides = "collect") +
  plot_annotation(tag_levels = "a") &
  theme(plot.tag = element_text(size = 22),
        plot.title = element_text(size = 18)) 

```


Boxes represent the 1.5 interquartile range, with the internal solid line representing the median. Circles represent data points. Difference in the alpha diversity indexes among conditions were tested using pairwise t-tests; p < 0.05 was considered the threshold significance for a difference between conditions.


# Who??

Which taxa were important? Which taxa were contributing to differences in community compositon?

**Note: Recovered 16S rRNA gene copy numbers do not equal organism abundance.**

That said, we can generate a heat map of our samples showing us how the relative abundance of different taxonomic groups change...potentially giving us a visual of which taxa are most important to the alpha and beta diversity patterns we observed. 
First, we're going to generate a custom table that will be easier to work with than a phyloseq object.



## Generate relative abundances

Our data currently shows number gene copies recovered, so we'll convert to percentages (relative abundances)


```{r}
data <- ps_min %>% 
  psmelt() %>% 
  select(-c(doc:sd_bp, sd_ba:bcd)) %>% 
  left_join(., gene_copies %>% 
              select(Phylum, Class, Order, copy_num)) %>% 
  arrange(Cruise, Station, datetime, dh) %>% 
  # mutate(interval = ifelse(Cruise == "NAAMES2" & dh == "Euphotic", 75, 100),
  #        interval = ifelse(Cruise == "NAAMES 3" & dh == "Euphotic", 50, interval)) %>% 
  group_by(Cruise, Station, datetime, dh) %>% 
  dplyr::rename(copies = Abundance) %>% 
  mutate(total_copies_dh = sum(copies, na.rm = T),
         mean_ba_dh = mean(ba, na.rm = T)) %>% 
  ungroup() %>% 
  group_by(Cruise, Station, datetime, dh, Family) %>% 
  mutate(fam_copies_dh = sum(copies, na.rm = T),
         fam_relabund_dh = fam_copies_dh/total_copies_dh, 
         fam_relacount_dh = fam_relabund_dh * mean_ba_dh, 
         fam_normcount_dh = fam_relacount_dh/copy_num) %>% 
  ungroup() %>% 
  select(Cruise:z, Sample, OTU, Kingdom:Genus, copies, total_copies_dh:fam_normcount_dh, everything())
```


```{r fig.height = 18, fig.width = 12}
relabund.plot <- data %>% 
  filter(Cruise == "NAAMES 2") %>% 
  filter(fam_relabund_dh >= 0.001) %>% 
  select(dh, plot_date, Family, fam_relabund_dh) %>% 
  distinct() %>% 
  mutate(Family = ifelse(Family == "NA", "Unassigned", Family)) %>% 
  ggplot(aes(x = plot_date, y = reorder(Family, fam_relabund_dh))) +
  geom_tile(aes(fill = fam_relabund_dh), color = "white") +
  scale_fill_viridis_c(trans = 'log10') +
  # scale_fill_viridis_b(option = "D",  trans = 'log10') +
  geom_text(aes(label = round(fam_relabund_dh, 3), color = "black"), size = 4) +
  scale_color_manual(values = c("white" = "white", "black" = "black")) +
  labs(x = "", y = "Family", fill = "Relative Abundance") +
  facet_grid(~factor(dh, levels = c("Euphotic", "Upper Mesopelagic"))) +
  theme_linedraw(base_size = 16) +
  theme(axis.text.y = element_text(size = 12), legend.position = "top") +
   guides(fill = guide_colourbar(barheight = 2, barwidth = 20, frame.colour = "black", frame.linewidth = 2,ticks.colour = "black", ticks.linewidth = 1), color = "none") +
  ggtitle("")
```


```{r fig.height = 18, fig.width = 12}
count.plot <- data %>% 
  filter(Cruise == "NAAMES 2") %>% 
  filter(fam_relabund_dh >= 0.001) %>% 
  filter(fam_normcount_dh > 0) %>% 
  select(dh, plot_date, Family, fam_relabund_dh, fam_normcount_dh) %>% 
  mutate(Family = ifelse(Family == "NA", "Unassigned", Family)) %>% 
  filter(!Family == "Unassigned") %>% #don't know what good copy numbers would be
  distinct() %>% 
  ggplot(aes(x = plot_date, y = reorder(Family, fam_relabund_dh))) +
  geom_tile(aes(fill = fam_normcount_dh), color = "white") +
  scale_fill_viridis_c(trans = 'log10') +
  # scale_fill_viridis_b(option = "D",  trans = 'log10') +
  geom_text(aes(label = formatC(fam_normcount_dh, format = "e", digits = 1), color = "black"), size = 4) +
  scale_color_manual(values = c("white" = "white", "black" = "black")) +
  labs(x = "", y = "Family", fill = expression(paste("Cells L"^-1))) +
  facet_grid(~factor(dh, levels = c("Euphotic", "Upper Mesopelagic"))) +
  theme_linedraw(base_size = 16) +
  theme(axis.text.y = element_text(size = 12), legend.position = "top") +
   guides(fill = guide_colourbar(barheight = 2, barwidth = 20, frame.colour = "black", frame.linewidth = 2,ticks.colour = "black", ticks.linewidth = 1), color = "none") +
  ggtitle("")
```


```{r fig.height=16, fig.width=22, message=FALSE, warning=FALSE}
relabund.plot + count.plot + plot_annotation(tag_levels = "a") 
```


```{r fig.height = 18, fig.width = 12}
n3_relabund.plot <- data %>% 
  filter(Cruise == "NAAMES 3") %>% 
  filter(fam_relabund_dh >= 0.001) %>% 
  select(dh, Station, plot_date, Family, fam_relabund_dh) %>% 
  mutate_at(vars(Station), as.character) %>% 
  distinct() %>% 
  mutate(Family = ifelse(Family == "NA", "Unassigned", Family)) %>% 
  ggplot(aes(x = Station, y = reorder(Family, fam_relabund_dh))) +
  geom_tile(aes(fill = fam_relabund_dh), color = "white") +
  scale_fill_viridis_c(trans = 'log10') +
  # scale_fill_viridis_b(option = "D",  trans = 'log10') +
  geom_text(aes(label = round(fam_relabund_dh, 3), color = "black"), size = 4) +
  scale_color_manual(values = c("white" = "white", "black" = "black")) +
  labs(x = "Station", y = "Family", fill = "Relative Abundance") +
  facet_grid(~factor(dh, levels = c("Euphotic", "Upper Mesopelagic"))) +
  theme_linedraw(base_size = 16) +
  theme(axis.text.y = element_text(size = 12), legend.position = "top") +
   guides(fill = guide_colourbar(barheight = 2, barwidth = 20, frame.colour = "black", frame.linewidth = 2,ticks.colour = "black", ticks.linewidth = 1), color = "none") +
  ggtitle("")
```


```{r fig.height = 18, fig.width = 12}
n3_count.plot <- data %>% 
  filter(Cruise == "NAAMES 3") %>% 
  filter(fam_relabund_dh >= 0.001) %>% 
  filter(fam_normcount_dh > 0) %>% 
  select(dh, plot_date, Station, Family, fam_relabund_dh, fam_normcount_dh) %>% 
  mutate_at(vars(Station), as.character) %>% 
  mutate(Family = ifelse(Family == "NA", "Unassigned", Family)) %>% 
  filter(!Family == "Unassigned") %>% #don't know what good copy numbers would be
  distinct() %>% 
  ggplot(aes(x = Station, y = reorder(Family, fam_relabund_dh))) +
  geom_tile(aes(fill = fam_normcount_dh), color = "white") +
  scale_fill_viridis_c(trans = 'log10') +
  # scale_fill_viridis_b(option = "D",  trans = 'log10') +
  geom_text(aes(label = formatC(fam_normcount_dh, format = "e", digits = 1), color = "black"), size = 4) +
  scale_color_manual(values = c("white" = "white", "black" = "black")) +
  labs(x = "Station", y = "Family", fill = expression(paste("Cells L"^-1))) +
  facet_grid(~factor(dh, levels = c("Euphotic", "Upper Mesopelagic"))) +
  theme_linedraw(base_size = 16) +
  theme(axis.text.y = element_text(size = 12), legend.position = "top") +
   guides(fill = guide_colourbar(barheight = 2, barwidth = 20, frame.colour = "black", frame.linewidth = 2,ticks.colour = "black", ticks.linewidth = 1), color = "none") +
  ggtitle("")
```


```{r fig.height=16, fig.width=22, message=FALSE, warning=FALSE}
n3_relabund.plot + n3_count.plot + plot_annotation(tag_levels = "a") 
```


```{r fig.height=18, fig.width=22, message=FALSE, warning=FALSE}
(data %>% 
  filter(Cruise == "NAAMES 2") %>% 
  filter(dh == "Upper Mesopelagic") %>%
  filter(fam_relabund_dh >= 0.001) %>% 
  filter(fam_normcount_dh > 0) %>% 
  select(dh, plot_date, Family, fam_relabund_dh, fam_normcount_dh) %>% 
  mutate(Family = ifelse(Family == "NA", "Unassigned", Family)) %>% 
  filter(!Family == "Unassigned") %>% #don't know what good copy numbers would be
  distinct() %>% 
  ggplot(aes(x = plot_date, y = reorder(Family, fam_relabund_dh))) +
  geom_tile(aes(fill = fam_normcount_dh), color = "white") +
  scale_fill_viridis_c(trans = 'log10') +
  # scale_fill_viridis_b(option = "D",  trans = 'log10') +
  geom_text(aes(label = formatC(fam_normcount_dh, format = "e", digits = 1), color = "black"), size = 4) +
  scale_color_manual(values = c("white" = "white", "black" = "black")) +
  labs(x = "", y = "Family", fill = expression(paste("Cells L"^-1))) +
  facet_grid(~factor(dh, levels = c("Euphotic", "Upper Mesopelagic"))) +
  theme_linedraw(base_size = 16) +
  theme(axis.text.y = element_text(size = 12), legend.position = "top") +
  guides(fill = guide_colourbar(barheight = 2, barwidth = 20, frame.colour = "black", frame.linewidth = 2,ticks.colour = "black", ticks.linewidth = 1), color = "none") +
  ggtitle("") ) +


(data %>% 
  filter(Cruise == "NAAMES 2") %>% 
  filter(dh == "Upper Mesopelagic") %>%
  filter(fam_relabund_dh >= 0.001) %>% 
  select(dh, plot_date, datetime, decimaldate, Phylum, Order, Family, fam_relabund_dh, fam_normcount_dh) %>% 
  mutate(Family = ifelse(Family == "NA", "Unassigned", Family)) %>% 
  filter(!Family == "Unassigned") %>% #don't know what good copy numbers would be
  distinct() %>% 
  ggplot(aes(y = reorder(Family, fam_relabund_dh), x = fam_normcount_dh, fill = Phylum)) +
  ggridges::geom_density_ridges2(scale = 3, alpha = 0.7, rel_min_height = 0.01) +
  labs(x = expression(paste("Log Cells L"^-1)), y = "Family", fill = expression(paste("Phylum"))) +
  scale_x_continuous(trans = 'log10') +
  facet_grid(~factor(dh, levels = c("Euphotic", "Upper Mesopelagic"))) +
  scale_fill_viridis_d() +
  theme_linedraw(base_size = 16)) 

```

# Identify "typical"surface and mesopelagic microbes

We'll make our designation based on our well-stratified profiles from NAAMES 3. 

```{r}
surf_bugs <- data %>% 
  filter(Cruise == "NAAMES 3", !Station == 3) %>% 
  filter(!dh == "Upper Mesopelagic") %>%
  filter(fam_relabund_dh > 0.01) %>%
  select(dh, datetime, Phylum, Order, Family, fam_relabund_dh, fam_normcount_dh) %>% 
  mutate(Family = ifelse(Family == "NA", "Unassigned", Family)) %>% 
  filter(!Family == "Unassigned") %>% #don't know what good copy numbers would be
  distinct() %>% 
  select(Family) %>% 
  dplyr::rename(surf = Family) %>% 
  as_vector()


meso_bugs <- data %>% 
              filter(Cruise == "NAAMES 3") %>% 
              filter(dh == "Upper Mesopelagic") %>%
              filter(fam_relabund_dh > 0.01) %>%
              select(dh, datetime, Phylum, Order, Family, fam_relabund_dh, fam_normcount_dh) %>% 
              mutate(Family = ifelse(Family == "NA", "Unassigned", Family)) %>% 
              filter(!Family == "Unassigned") %>% #don't know what good copy numbers would be
              distinct() %>% 
              select(Family) %>% 
              dplyr::rename(meso = Family) %>% 
  as_vector()

unique_surf_bugs <- setdiff(surf_bugs, meso_bugs)

unique_meso_bugs <- setdiff(meso_bugs, surf_bugs)
```

# Taxa changes over time

```{r}
s4_meso_subset <- data %>% 
  filter(Cruise == "NAAMES 2") %>% 
  filter(dh == "Upper Mesopelagic") %>%
  filter(fam_relabund_dh >= 0.001) %>% 
  filter(Family %in% c(unique_meso_bugs) | Family == "SAR11_II") %>% 
  select(dh, datetime, Phylum, Order, Family, Genus, fam_relabund_dh, fam_normcount_dh) %>% 
  mutate(Family = ifelse(Family == "NA", "Unassigned", Family)) %>% 
  filter(!Family == "Unassigned") %>% #don't know what good copy numbers would be
  distinct()  
  
```


```{r fig.height=12, fig.width= 26, message=FALSE, warning=FALSE}
mesobugs.plot <- s4_meso_subset %>% 
  filter(!Family == "SAR202_Clade2") %>% #only appears once
  ggplot(aes(x = datetime, y = fam_relabund_dh)) +
  facet_wrap(~Family, dir = "v", scales = "free_y") +
  geom_line(size = 0.7) +
  geom_point(shape = 21, size = 4, fill = "white", color = "black", stroke = 1) + 
  geom_line(data = s4_meso_subset %>% filter(!Family == "SAR202_Clade2"), aes(x = datetime, y = fam_normcount_dh/10^9), color = "#377EB8") +
  geom_point(data = s4_meso_subset %>% filter(!Family == "SAR202_Clade2"), aes(x = datetime, y = fam_normcount_dh/10^9), shape = 21, size = 6, fill = "#377EB8", color = "black", stroke = 1) + 
  scale_y_continuous(
    # Add a second axis and specify its features
    sec.axis = sec_axis(~.*10^9, name = expression(paste("Cell Abundance, L"^-1)))
  ) + 
  labs(x = "", y = expression(paste("Relative Abundance, %")), colour = "", fill = "") +
  theme_linedraw(base_size = 16) +
  theme(axis.title.y.right = element_text(color = "#377EB8"),
        axis.text.y.right = element_text(color = "#377EB8"))
```

```{r}
s4_meso_subset_ez <- data %>% 
  filter(Cruise == "NAAMES 2") %>% 
  filter(dh == "Upper Mesopelagic") %>%
  filter(fam_relabund_dh >= 0.001) %>% 
  filter(Family %in% c(unique_surf_bugs) | Family == "SAR11_Ia") %>% 
  select(dh, datetime, Phylum, Order, Family, Genus, fam_relabund_dh, fam_normcount_dh) %>% 
  mutate(Family = ifelse(Family == "NA", "Unassigned", Family)) %>% 
  filter(!Family == "Unassigned") %>% #don't know what good copy numbers would be
  distinct()  
  
```

```{r}
s4_meso_subset %>% select(Family, Genus) %>% 
  mutate(name = paste(Family, Genus, sep = "_")) %>% 
  select(name) %>% 
  distinct()
```


```{r fig.height=12, fig.width= 26, message=FALSE, warning=FALSE}
surfbugs.plot <- s4_meso_subset_ez %>% 
  ggplot(aes(x = datetime, y = fam_relabund_dh)) +
  facet_wrap(~Family, dir = "v", scales = "free_y") +
  geom_line(size = 0.7) +
  geom_point(shape = 21, size = 4, fill = "white", color = "black", stroke = 1) + 
  geom_line(data = s4_meso_subset_ez, aes(x = datetime, y = fam_normcount_dh/10^9), color = "#377EB8") +
  geom_point(data = s4_meso_subset_ez, aes(x = datetime, y = fam_normcount_dh/10^9), shape = 21, size = 6, fill = "#377EB8", color = "black", stroke = 1) +
  scale_y_continuous(
    # Add a second axis and specify its features
    sec.axis = sec_axis(~.*10^9, name = expression(paste("Cell Abundance, L"^-1)))
  ) + 
  labs(x = "", y = expression(paste("Relative Abundance, %")), colour = "", fill = "") +
  theme_linedraw(base_size = 16) +
  theme(axis.title.y.right = element_text(color = "#377EB8"),
        axis.text.y.right = element_text(color = "#377EB8"))
```


```{r}
s4_meso_subset_ez %>% select(Family, Genus) %>% 
  mutate(name = paste(Family, Genus, sep = "_")) %>% 
  select(name) %>% 
  distinct()
```


```{r fig.height=17, fig.width=22, message=TRUE, warning=FALSE}
surfbugs.plot / mesobugs.plot + plot_annotation(tag_levels = "a") 
```



```{r }
levels <- c("SAR11_Ia", "Flavobacteriaceae", "ZD0405", "SAR86_clade", "SAR11_II", "SAR202_Clade1", "OM1_clade", "Salinisphaeraceae", "OCS116_clade")
```


```{r}
write_rds(data, "~/GITHUB/naames_multiday/Output/processed_16S.rds")
```









